<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <title>report.html</title>
    <link href="assets/style.css" rel="stylesheet" type="text/css"/></head>
  <body onLoad="init()">
    <script>/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */


function toArray(iter) {
    if (iter === null) {
        return null;
    }
    return Array.prototype.slice.call(iter);
}

function find(selector, elem) { // eslint-disable-line no-redeclare
    if (!elem) {
        elem = document;
    }
    return elem.querySelector(selector);
}

function findAll(selector, elem) {
    if (!elem) {
        elem = document;
    }
    return toArray(elem.querySelectorAll(selector));
}

function sortColumn(elem) {
    toggleSortStates(elem);
    const colIndex = toArray(elem.parentNode.childNodes).indexOf(elem);
    let key;
    if (elem.classList.contains('result')) {
        key = keyResult;
    } else if (elem.classList.contains('links')) {
        key = keyLink;
    } else {
        key = keyAlpha;
    }
    sortTable(elem, key(colIndex));
}

function showAllExtras() { // eslint-disable-line no-unused-vars
    findAll('.col-result').forEach(showExtras);
}

function hideAllExtras() { // eslint-disable-line no-unused-vars
    findAll('.col-result').forEach(hideExtras);
}

function showExtras(colresultElem) {
    const extras = colresultElem.parentNode.nextElementSibling;
    const expandcollapse = colresultElem.firstElementChild;
    extras.classList.remove('collapsed');
    expandcollapse.classList.remove('expander');
    expandcollapse.classList.add('collapser');
}

function hideExtras(colresultElem) {
    const extras = colresultElem.parentNode.nextElementSibling;
    const expandcollapse = colresultElem.firstElementChild;
    extras.classList.add('collapsed');
    expandcollapse.classList.remove('collapser');
    expandcollapse.classList.add('expander');
}

function showFilters() {
    let visibleString = getQueryParameter('visible') || 'all';
    visibleString = visibleString.toLowerCase();
    const checkedItems = visibleString.split(',');

    const filterItems = document.getElementsByClassName('filter');
    for (let i = 0; i < filterItems.length; i++) {
        filterItems[i].hidden = false;

        if (visibleString != 'all') {
            filterItems[i].checked = checkedItems.includes(filterItems[i].getAttribute('data-test-result'));
            filterTable(filterItems[i]);
        }
    }
}

function addCollapse() {
    // Add links for show/hide all
    const resulttable = find('table#results-table');
    const showhideall = document.createElement('p');
    showhideall.innerHTML = '<a href="javascript:showAllExtras()">Show all details</a> / ' +
                            '<a href="javascript:hideAllExtras()">Hide all details</a>';
    resulttable.parentElement.insertBefore(showhideall, resulttable);

    // Add show/hide link to each result
    findAll('.col-result').forEach(function(elem) {
        const collapsed = getQueryParameter('collapsed') || 'Passed';
        const extras = elem.parentNode.nextElementSibling;
        const expandcollapse = document.createElement('span');
        if (extras.classList.contains('collapsed')) {
            expandcollapse.classList.add('expander');
        } else if (collapsed.includes(elem.innerHTML)) {
            extras.classList.add('collapsed');
            expandcollapse.classList.add('expander');
        } else {
            expandcollapse.classList.add('collapser');
        }
        elem.appendChild(expandcollapse);

        elem.addEventListener('click', function(event) {
            if (event.currentTarget.parentNode.nextElementSibling.classList.contains('collapsed')) {
                showExtras(event.currentTarget);
            } else {
                hideExtras(event.currentTarget);
            }
        });
    });
}

function getQueryParameter(name) {
    const match = RegExp('[?&]' + name + '=([^&]*)').exec(window.location.search);
    return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
}

function init () { // eslint-disable-line no-unused-vars
    resetSortHeaders();

    addCollapse();

    showFilters();

    sortColumn(find('.initial-sort'));

    findAll('.sortable').forEach(function(elem) {
        elem.addEventListener('click',
            function() {
                sortColumn(elem);
            }, false);
    });
}

function sortTable(clicked, keyFunc) {
    const rows = findAll('.results-table-row');
    const reversed = !clicked.classList.contains('asc');
    const sortedRows = sort(rows, keyFunc, reversed);
    /* Whole table is removed here because browsers acts much slower
     * when appending existing elements.
     */
    const thead = document.getElementById('results-table-head');
    document.getElementById('results-table').remove();
    const parent = document.createElement('table');
    parent.id = 'results-table';
    parent.appendChild(thead);
    sortedRows.forEach(function(elem) {
        parent.appendChild(elem);
    });
    document.getElementsByTagName('BODY')[0].appendChild(parent);
}

function sort(items, keyFunc, reversed) {
    const sortArray = items.map(function(item, i) {
        return [keyFunc(item), i];
    });

    sortArray.sort(function(a, b) {
        const keyA = a[0];
        const keyB = b[0];

        if (keyA == keyB) return 0;

        if (reversed) {
            return keyA < keyB ? 1 : -1;
        } else {
            return keyA > keyB ? 1 : -1;
        }
    });

    return sortArray.map(function(item) {
        const index = item[1];
        return items[index];
    });
}

function keyAlpha(colIndex) {
    return function(elem) {
        return elem.childNodes[1].childNodes[colIndex].firstChild.data.toLowerCase();
    };
}

function keyLink(colIndex) {
    return function(elem) {
        const dataCell = elem.childNodes[1].childNodes[colIndex].firstChild;
        return dataCell == null ? '' : dataCell.innerText.toLowerCase();
    };
}

function keyResult(colIndex) {
    return function(elem) {
        const strings = ['Error', 'Failed', 'Rerun', 'XFailed', 'XPassed',
            'Skipped', 'Passed'];
        return strings.indexOf(elem.childNodes[1].childNodes[colIndex].firstChild.data);
    };
}

function resetSortHeaders() {
    findAll('.sort-icon').forEach(function(elem) {
        elem.parentNode.removeChild(elem);
    });
    findAll('.sortable').forEach(function(elem) {
        const icon = document.createElement('div');
        icon.className = 'sort-icon';
        icon.textContent = 'vvv';
        elem.insertBefore(icon, elem.firstChild);
        elem.classList.remove('desc', 'active');
        elem.classList.add('asc', 'inactive');
    });
}

function toggleSortStates(elem) {
    //if active, toggle between asc and desc
    if (elem.classList.contains('active')) {
        elem.classList.toggle('asc');
        elem.classList.toggle('desc');
    }

    //if inactive, reset all other functions and add ascending active
    if (elem.classList.contains('inactive')) {
        resetSortHeaders();
        elem.classList.remove('inactive');
        elem.classList.add('active');
    }
}

function isAllRowsHidden(value) {
    return value.hidden == false;
}

function filterTable(elem) { // eslint-disable-line no-unused-vars
    const outcomeAtt = 'data-test-result';
    const outcome = elem.getAttribute(outcomeAtt);
    const classOutcome = outcome + ' results-table-row';
    const outcomeRows = document.getElementsByClassName(classOutcome);

    for(let i = 0; i < outcomeRows.length; i++){
        outcomeRows[i].hidden = !elem.checked;
    }

    const rows = findAll('.results-table-row').filter(isAllRowsHidden);
    const allRowsHidden = rows.length == 0 ? true : false;
    const notFoundMessage = document.getElementById('not-found-message');
    notFoundMessage.hidden = !allRowsHidden;
}
</script>
    <h1>report.html</h1>
    <p>Report generated on 15-Sep-2023 at 07:48:49 by <a href="https://pypi.python.org/pypi/pytest-html">pytest-html</a> v3.2.0</p>
    <h2>Summary</h2>
    <p>8 tests ran in 30.58 seconds. </p>
    <p class="filter" hidden="true">(Un)check the boxes to filter the results.</p><input checked="true" class="filter" data-test-result="passed" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="passed">8 passed</span>, <input checked="true" class="filter" data-test-result="skipped" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="skipped">0 skipped</span>, <input checked="true" class="filter" data-test-result="failed" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="failed">0 failed</span>, <input checked="true" class="filter" data-test-result="error" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="error">0 errors</span>, <input checked="true" class="filter" data-test-result="xfailed" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="xfailed">0 expected failures</span>, <input checked="true" class="filter" data-test-result="xpassed" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="xpassed">0 unexpected passes</span>
    <h2>Results</h2>
    <table id="results-table">
      <thead id="results-table-head">
        <tr>
          <th class="sortable result initial-sort" col="result">Result</th>
          <th class="sortable" col="name">Test</th>
          <th class="sortable" col="duration">Duration</th>
          <th class="sortable links" col="links">Links</th></tr>
        <tr hidden="true" id="not-found-message">
          <th colspan="4">No results found. Try to check the filters</th></tr></thead>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">biobb_model/test/unitests/test_model/test_checking_log.py::TestCheckingLog::test_launch</td>
          <td class="col-duration">17.89</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>2023-09-15 07:48:19,681 [MainThread  ] [INFO ]  Executing biobb_model.model.checking_log Version: 4.1.0
2023-09-15 07:48:19,682 [MainThread  ] [INFO ]  Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_11/checking_log/f5cf4d36-d2ba-4a8f-8319-168ac395e69c
2023-09-15 07:48:19,682 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_11/checking_log/f5cf4d36-d2ba-4a8f-8319-168ac395e69c/2ki5.pdb checkall &gt; /tmp/biobb/unitests_11/checking_log/f5cf4d36-d2ba-4a8f-8319-168ac395e69c/output_log_path.log

2023-09-15 07:48:37,553 [MainThread  ] [INFO ]  Exit code 0

2023-09-15 07:48:37,553 [MainThread  ] [INFO ]  Removed: [&#x27;/tmp/biobb/unitests_11/checking_log/f5cf4d36-d2ba-4a8f-8319-168ac395e69c&#x27;]
Checking if empty file: /tmp/biobb/unitests_11/checking_log/output_log_path.log
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     /tmp/biobb/unitests_11/checking_log/checking_log_log.out:file_utils.py:390 Executing biobb_model.model.checking_log Version: 4.1.0
INFO     /tmp/biobb/unitests_11/checking_log/checking_log_log.out:file_utils.py:390 Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_11/checking_log/f5cf4d36-d2ba-4a8f-8319-168ac395e69c
INFO     /tmp/biobb/unitests_11/checking_log/checking_log_log.out:cmd_wrapper.py:33 check_structure -i /tmp/biobb/unitests_11/checking_log/f5cf4d36-d2ba-4a8f-8319-168ac395e69c/2ki5.pdb checkall &gt; /tmp/biobb/unitests_11/checking_log/f5cf4d36-d2ba-4a8f-8319-168ac395e69c/output_log_path.log

INFO     /tmp/biobb/unitests_11/checking_log/checking_log_log.out:cmd_wrapper.py:50 Exit code 0

INFO     /tmp/biobb/unitests_11/checking_log/checking_log_log.out:file_utils.py:390 Removed: [&#x27;/tmp/biobb/unitests_11/checking_log/f5cf4d36-d2ba-4a8f-8319-168ac395e69c&#x27;]<br/> ----------------------------Captured stdout teardown---------------------------- <br/>
Removing: /tmp/biobb/unitests_11
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">biobb_model/test/unitests/test_model/test_fix_altlocs.py::TestFixAltlocs::test_launch</td>
          <td class="col-duration">1.15</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>2023-09-15 07:48:37,582 [MainThread  ] [INFO ]  Executing biobb_model.model.fix_altlocs Version: 4.1.0
2023-09-15 07:48:37,583 [MainThread  ] [INFO ]  Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/3ebp.pdb to /tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7
2023-09-15 07:48:37,583 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7/3ebp.pdb -o /tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7/output_pdb_path.pdb --force_save --non_interactive altloc --select A339:A,A171:B,A768:A

2023-09-15 07:48:38,397 [MainThread  ] [INFO ]  Exit code 0

2023-09-15 07:48:38,397 [MainThread  ] [INFO ]  ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7/3ebp.pdb loaded
 PDB id: 3EBP 
 Title: glycogen phosphorylase b/flavopiridol complex
 Experimental method: x-ray diffraction
 Keywords: glycogen phosphorylase, rational inhibitor design, glycogenolysis, diabetes type 2, allosteric enzyme, carbohydrate metabolism, glycogen metabolism, glycosyltransferase, nucleotide-binding, phosphoprotein, pyridoxal phosphate, transferase-transferase inhibitor complex
 Resolution (A): 2.0

 Num. models: 1
 Num. chains: 1 (A: Protein)
 Num. residues:  1097
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  288
 Num. ligands or modified residues:  2
 Num. water mol.:  286
 Num. atoms:  6918
Modified residues found
LLP A680
Small mol ligands found
CPB A940
Running altloc. Options: --select A339:A,A171:B,A768:A
Detected 3 residues with alternative location labels
CYS A171
  CB   A (0.50) B (0.50)
  SG   A (0.50) B (0.50)
ASP A339
  CG   A (0.50) B (0.50)
  OD1  A (0.50) B (0.50)
  OD2  A (0.50) B (0.50)
HIS A768
  CG   A (0.50) B (0.50)
  ND1  A (0.50) B (0.50)
  CD2  A (0.50) B (0.50)
  CE1  A (0.50) B (0.50)
  NE2  A (0.50) B (0.50)
Selecting location A339:A,A171:B,A768:A
Final Num. models: 1
Final Num. chains: 1 (A: Protein)
Final Num. residues:  1097
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  288
Final Num. ligands or modified residues:  2
Final Num. water mol.:  286
Final Num. atoms:  6918
Modified residues found
LLP A680
Small mol ligands found
CPB A940
Structure saved on /tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7/output_pdb_path.pdb

2023-09-15 07:48:38,399 [MainThread  ] [INFO ]  Removed: [&#x27;/tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7&#x27;]
Checking if empty file: /tmp/biobb/unitests_11/fix_altlocs/output_pdb_path.pdb
Checking RMSD between:
     PDB_A: /tmp/biobb/unitests_11/fix_altlocs/output_pdb_path.pdb
     PDB_B: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_altloc.pdb
     Ignoring HETAMT in RMSD
     Ignoring Hydrogen atoms in RMSD
     Atoms ALIGNED in PDB_A: 6866
     Atoms ALIGNED in PDB_B: 6866
     RMS: 1.0683071939348411e-14
     RMS_CUTOFF: 1
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     /tmp/biobb/unitests_11/fix_altlocs/fix_altlocs_log.out:file_utils.py:390 Executing biobb_model.model.fix_altlocs Version: 4.1.0
INFO     /tmp/biobb/unitests_11/fix_altlocs/fix_altlocs_log.out:file_utils.py:390 Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/3ebp.pdb to /tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7
INFO     /tmp/biobb/unitests_11/fix_altlocs/fix_altlocs_log.out:cmd_wrapper.py:33 check_structure -i /tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7/3ebp.pdb -o /tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7/output_pdb_path.pdb --force_save --non_interactive altloc --select A339:A,A171:B,A768:A

INFO     /tmp/biobb/unitests_11/fix_altlocs/fix_altlocs_log.out:cmd_wrapper.py:50 Exit code 0

INFO     /tmp/biobb/unitests_11/fix_altlocs/fix_altlocs_log.out:cmd_wrapper.py:52 ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7/3ebp.pdb loaded
 PDB id: 3EBP 
 Title: glycogen phosphorylase b/flavopiridol complex
 Experimental method: x-ray diffraction
 Keywords: glycogen phosphorylase, rational inhibitor design, glycogenolysis, diabetes type 2, allosteric enzyme, carbohydrate metabolism, glycogen metabolism, glycosyltransferase, nucleotide-binding, phosphoprotein, pyridoxal phosphate, transferase-transferase inhibitor complex
 Resolution (A): 2.0

 Num. models: 1
 Num. chains: 1 (A: Protein)
 Num. residues:  1097
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  288
 Num. ligands or modified residues:  2
 Num. water mol.:  286
 Num. atoms:  6918
Modified residues found
LLP A680
Small mol ligands found
CPB A940
Running altloc. Options: --select A339:A,A171:B,A768:A
Detected 3 residues with alternative location labels
CYS A171
  CB   A (0.50) B (0.50)
  SG   A (0.50) B (0.50)
ASP A339
  CG   A (0.50) B (0.50)
  OD1  A (0.50) B (0.50)
  OD2  A (0.50) B (0.50)
HIS A768
  CG   A (0.50) B (0.50)
  ND1  A (0.50) B (0.50)
  CD2  A (0.50) B (0.50)
  CE1  A (0.50) B (0.50)
  NE2  A (0.50) B (0.50)
Selecting location A339:A,A171:B,A768:A
Final Num. models: 1
Final Num. chains: 1 (A: Protein)
Final Num. residues:  1097
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  288
Final Num. ligands or modified residues:  2
Final Num. water mol.:  286
Final Num. atoms:  6918
Modified residues found
LLP A680
Small mol ligands found
CPB A940
Structure saved on /tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7/output_pdb_path.pdb

INFO     /tmp/biobb/unitests_11/fix_altlocs/fix_altlocs_log.out:file_utils.py:390 Removed: [&#x27;/tmp/biobb/unitests_11/fix_altlocs/e6e6febe-896e-4de2-a289-10332f6ed5c7&#x27;]<br/> ----------------------------Captured stdout teardown---------------------------- <br/>
Removing: /tmp/biobb/unitests_11
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">biobb_model/test/unitests/test_model/test_fix_amides.py::TestFixAmides::test_launch</td>
          <td class="col-duration">1.85</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>2023-09-15 07:48:38,738 [MainThread  ] [INFO ]  Executing biobb_model.model.fix_amides Version: 4.1.0
2023-09-15 07:48:38,739 [MainThread  ] [INFO ]  Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/5s2z.pdb to /tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9
2023-09-15 07:48:38,740 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9/5s2z.pdb -o /tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9/output_pdb_path.pdb --force_save amide --fix All

2023-09-15 07:48:40,442 [MainThread  ] [INFO ]  Exit code 0

2023-09-15 07:48:40,443 [MainThread  ] [INFO ]  ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9/5s2z.pdb loaded
 PDB id: 5S2Z 
 Title: pandda analysis group deposition -- crystal structure of sars-cov-2 nsp3 macrodomain in complex with z126932614
 Experimental method: x-ray diffraction
 Keywords: sgc - diamond i04-1 fragment screening, pandda, xchemexplorer, sars- cov-2 nsp3 macrodomain, viral protein
 Resolution (A): 1.07

 Num. models: 1
 Num. chains: 2 (A: Protein, B: Protein)
 Num. residues:  792
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  458
 Num. ligands or modified residues:  1
 Num. water mol.:  457
 Num. atoms:  2998
Modified residues found
T6J A401
Running amide. Options: --fix All
Found 2 unusual contact(s) involving amide atoms
 ASN B20.OD1  VAL B155.O      3.015 A
 ASN B20.OD1  PHE B156.O      2.894 A
Amide residues fixed All (1)
Rechecking
Found no unusual contact(s) involving amide atoms
Final Num. models: 1
Final Num. chains: 2 (A: Protein, B: Protein)
Final Num. residues:  792
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  458
Final Num. ligands or modified residues:  1
Final Num. water mol.:  457
Final Num. atoms:  2998
Modified residues found
T6J A401
Structure saved on /tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9/output_pdb_path.pdb

2023-09-15 07:48:40,444 [MainThread  ] [INFO ]  Removed: [&#x27;/tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9&#x27;]
Checking if empty file: /tmp/biobb/unitests_11/fix_amides/output_pdb_path.pdb
Checking RMSD between:
     PDB_A: /tmp/biobb/unitests_11/fix_amides/output_pdb_path.pdb
     PDB_B: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_amide_pdb_path.pdb
     Ignoring HETAMT in RMSD
     Ignoring Hydrogen atoms in RMSD
     Atoms ALIGNED in PDB_A: 2984
     Atoms ALIGNED in PDB_B: 2984
     RMS: 7.386865969918367e-15
     RMS_CUTOFF: 1
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     /tmp/biobb/unitests_11/fix_amides/fix_amides_log.out:file_utils.py:390 Executing biobb_model.model.fix_amides Version: 4.1.0
INFO     /tmp/biobb/unitests_11/fix_amides/fix_amides_log.out:file_utils.py:390 Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/5s2z.pdb to /tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9
INFO     /tmp/biobb/unitests_11/fix_amides/fix_amides_log.out:cmd_wrapper.py:33 check_structure -i /tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9/5s2z.pdb -o /tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9/output_pdb_path.pdb --force_save amide --fix All

INFO     /tmp/biobb/unitests_11/fix_amides/fix_amides_log.out:cmd_wrapper.py:50 Exit code 0

INFO     /tmp/biobb/unitests_11/fix_amides/fix_amides_log.out:cmd_wrapper.py:52 ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9/5s2z.pdb loaded
 PDB id: 5S2Z 
 Title: pandda analysis group deposition -- crystal structure of sars-cov-2 nsp3 macrodomain in complex with z126932614
 Experimental method: x-ray diffraction
 Keywords: sgc - diamond i04-1 fragment screening, pandda, xchemexplorer, sars- cov-2 nsp3 macrodomain, viral protein
 Resolution (A): 1.07

 Num. models: 1
 Num. chains: 2 (A: Protein, B: Protein)
 Num. residues:  792
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  458
 Num. ligands or modified residues:  1
 Num. water mol.:  457
 Num. atoms:  2998
Modified residues found
T6J A401
Running amide. Options: --fix All
Found 2 unusual contact(s) involving amide atoms
 ASN B20.OD1  VAL B155.O      3.015 A
 ASN B20.OD1  PHE B156.O      2.894 A
Amide residues fixed All (1)
Rechecking
Found no unusual contact(s) involving amide atoms
Final Num. models: 1
Final Num. chains: 2 (A: Protein, B: Protein)
Final Num. residues:  792
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  458
Final Num. ligands or modified residues:  1
Final Num. water mol.:  457
Final Num. atoms:  2998
Modified residues found
T6J A401
Structure saved on /tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9/output_pdb_path.pdb

INFO     /tmp/biobb/unitests_11/fix_amides/fix_amides_log.out:file_utils.py:390 Removed: [&#x27;/tmp/biobb/unitests_11/fix_amides/f1e87559-5138-43c6-b893-fb26d116a3c9&#x27;]<br/> ----------------------------Captured stdout teardown---------------------------- <br/>
Removing: /tmp/biobb/unitests_11
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">biobb_model/test/unitests/test_model/test_fix_chirality.py::TestFixChirality::test_launch</td>
          <td class="col-duration">1.44</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>2023-09-15 07:48:40,597 [MainThread  ] [INFO ]  Executing biobb_model.model.fix_chirality Version: 4.1.0
2023-09-15 07:48:40,598 [MainThread  ] [INFO ]  Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/3e2y.pdb to /tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c
2023-09-15 07:48:40,598 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c/3e2y.pdb -o /tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c/output_pdb_path.pdb --force_save chiral --fix All

2023-09-15 07:48:41,662 [MainThread  ] [INFO ]  Exit code 0

2023-09-15 07:48:41,662 [MainThread  ] [INFO ]  ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c/3e2y.pdb loaded
 PDB id: 3E2Y 
 Title: crystal structure of mouse kynurenine aminotransferase iii in complex with glutamine
 Experimental method: x-ray diffraction
 Keywords: alpha beta protein, plp dependent protein, aminotransferase, lyase, pyridoxal phosphate, transferase
 Resolution (A): 2.26

 Num. models: 1
 Num. chains: 2 (A: Protein, B: Protein)
 Num. residues:  1239
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  419
 Num. ligands or modified residues:  8
 Num. water mol.:  411
 Num. atoms:  6993
Modified residues found
GLN A456
GLN B457
Small mol ligands found
PMP A457
GOL A458
GOL A459
PMP B456
GOL B458
GOL B459
Running chiral. Options: --fix All
Found 2 residues with incorrect side-chain chirality
 THR A256  
 THR B286  
Chiral side chains fixed All (2)
Checking for steric clashes
No severe clashes detected
1 Steric apolar clashes detected
 THR A256.CG2 GLU A348.OE2    2.678 A
No polar_acceptor clashes detected
1 Steric polar_donor clashes detected
 GLN A97.NE2  THR B286.N      3.001 A
No positive clashes detected
No negative clashes detected
Final Num. models: 1
Final Num. chains: 2 (A: Protein, B: Protein)
Final Num. residues:  1239
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  419
Final Num. ligands or modified residues:  8
Final Num. water mol.:  411
Final Num. atoms:  6993
Modified residues found
GLN A456
GLN B457
Small mol ligands found
PMP A457
GOL A458
GOL A459
PMP B456
GOL B458
GOL B459
Structure saved on /tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c/output_pdb_path.pdb

2023-09-15 07:48:41,663 [MainThread  ] [INFO ]  Removed: [&#x27;/tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c&#x27;]
Checking if empty file: /tmp/biobb/unitests_11/fix_chirality/output_pdb_path.pdb
Checking RMSD between:
     PDB_A: /tmp/biobb/unitests_11/fix_chirality/output_pdb_path.pdb
     PDB_B: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_chirality_pdb_path.pdb
     Ignoring HETAMT in RMSD
     Ignoring Hydrogen atoms in RMSD
     Atoms ALIGNED in PDB_A: 6917
     Atoms ALIGNED in PDB_B: 6917
     RMS: 6.2661067786844326e-15
     RMS_CUTOFF: 1
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     /tmp/biobb/unitests_11/fix_chirality/fix_chirality_log.out:file_utils.py:390 Executing biobb_model.model.fix_chirality Version: 4.1.0
INFO     /tmp/biobb/unitests_11/fix_chirality/fix_chirality_log.out:file_utils.py:390 Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/3e2y.pdb to /tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c
INFO     /tmp/biobb/unitests_11/fix_chirality/fix_chirality_log.out:cmd_wrapper.py:33 check_structure -i /tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c/3e2y.pdb -o /tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c/output_pdb_path.pdb --force_save chiral --fix All

INFO     /tmp/biobb/unitests_11/fix_chirality/fix_chirality_log.out:cmd_wrapper.py:50 Exit code 0

INFO     /tmp/biobb/unitests_11/fix_chirality/fix_chirality_log.out:cmd_wrapper.py:52 ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c/3e2y.pdb loaded
 PDB id: 3E2Y 
 Title: crystal structure of mouse kynurenine aminotransferase iii in complex with glutamine
 Experimental method: x-ray diffraction
 Keywords: alpha beta protein, plp dependent protein, aminotransferase, lyase, pyridoxal phosphate, transferase
 Resolution (A): 2.26

 Num. models: 1
 Num. chains: 2 (A: Protein, B: Protein)
 Num. residues:  1239
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  419
 Num. ligands or modified residues:  8
 Num. water mol.:  411
 Num. atoms:  6993
Modified residues found
GLN A456
GLN B457
Small mol ligands found
PMP A457
GOL A458
GOL A459
PMP B456
GOL B458
GOL B459
Running chiral. Options: --fix All
Found 2 residues with incorrect side-chain chirality
 THR A256  
 THR B286  
Chiral side chains fixed All (2)
Checking for steric clashes
No severe clashes detected
1 Steric apolar clashes detected
 THR A256.CG2 GLU A348.OE2    2.678 A
No polar_acceptor clashes detected
1 Steric polar_donor clashes detected
 GLN A97.NE2  THR B286.N      3.001 A
No positive clashes detected
No negative clashes detected
Final Num. models: 1
Final Num. chains: 2 (A: Protein, B: Protein)
Final Num. residues:  1239
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  419
Final Num. ligands or modified residues:  8
Final Num. water mol.:  411
Final Num. atoms:  6993
Modified residues found
GLN A456
GLN B457
Small mol ligands found
PMP A457
GOL A458
GOL A459
PMP B456
GOL B458
GOL B459
Structure saved on /tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c/output_pdb_path.pdb

INFO     /tmp/biobb/unitests_11/fix_chirality/fix_chirality_log.out:file_utils.py:390 Removed: [&#x27;/tmp/biobb/unitests_11/fix_chirality/59e636e9-2463-450c-96b0-efff16f2104c&#x27;]<br/> ----------------------------Captured stdout teardown---------------------------- <br/>
Removing: /tmp/biobb/unitests_11
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">biobb_model/test/unitests/test_model/test_fix_pdb.py::TestFixPdb::test_launch</td>
          <td class="col-duration">4.67</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>2023-09-15 07:48:42,039 [MainThread  ] [INFO ]  Executing biobb_model.model.fix_pdb Version: 4.1.0
126 VVIKVCEFQFCNDPFLGVYYHKNNKSWMESEFRVYSSANNCTFEYVSQPFLMDLEGKQGNFKNLREFVFKNIDGYFK-----IYSKH------TPINLVRDLPQGF-SALEPLVDLPIG----INIT------------------RFQTLLAL--------HRSYLT-PGDS------SSGWTAG------------------AAAYYVGYLQ------PRTFLLKYNENGTITDAVDCALDPLSETKCTLKSFTVEKGIYQTSNFRVQPTESIVRF-----PNITNLCPFGEVFNATRFASVYA------------WNRKR--------------------ISNCVADYSVL-YNSASFSTFKCYGV--------SPTKLNDLCFTNVYADSFVIRGDEVRQIAPGQTGKIADYNYKLPDDFTGCVIAWNSNNLDSKVGGNYNY---LYRLFRKSNLKPF
    ||.|       |....|..|......|.    |.|........||.......|.|......|.|    ......|..     .|..|      .|..|..|....| ..|.||. .|.|    |..|                  .|.....|        ..|.|| |||.      ...|..|                  .|....|..|      ....||....|.....||.........|...||..    |.. .|.| ....|....|     .||....||.......|. .|..            |..||                    ......|||.. |.......|.....        .|....|.       ......|....|..  ..||.......| .||.|      |.|.|  ||....|   ||......|.|.|
166 VVLK-------NEMARGYHYEDYGDYWR----RDYETEGSPDLEYSRDQLIKDVERIFAEIKPL----YEQLHAYVRTKLMDTYPFHISPTGCLPAHLLGDMWGRFWTNLYPLT-VPFGQKPNIDVTDAMLNQNWDAKRIFKEAEKFFVSIGLPNMTEGFWNNSMLTDPGDGRKVVCHPTAWDLGKGDFRIKMCTKVTMEDFLTAHHEMGHIQYDMAYASQPYLLRNGANEGFHEAVGEVMSLSVATPKHLKTM----GLL-SSDF-LEDNETEINFLFKQALNIVGTLPFTYMLEKWRW-MVFKGEIPKEEWMKKWWEMKRKIVGVVEPVPHDETYCDPASLFHVANDYSFIRYYTRTIFEFQFHEALCRIAKHDGPLHKCDI-------SNSTDAGEKLHQML--SVGKSQPWTSVL-KDFVG------SKNMD--VGPLLRYFEPLYTWLTEQNRKSF
  Score=67

Normalized score: 0.11185308848080133
Not valid alignment
19 STIEEQAKTFLDKFNHEAEDLFYQSSLASWNYNTNITEENVQNMNNAGDKWSAFLKEQSTLAQMYPLQEIQNLTVKLQLQALQQNGSSVLSEDKSKRLNTILNTMSTIYSTGKVCNPDNPQECLLLEPGLNEIMANSLDYNERLWAWESWRSEVGKQLRPLYEEYVVLKNEMARANHYEDYGDYWRGDYEVNGVDGYDYSRGQLIEDVEHTFEEIKPLYEHLHAYVRAKLMNAYPSYISPIGCLPAHLLGDMWGRFWTNLYSLTVPFGQKPNIDVTDAMVDQAWDAQRIFKEAEKFFVSVGLPNMTQGFWENSMLTDPGNVQKAVCHPTAWDLGKGDFRILMCTKVTMDDFLTAHHEMGHIQYDMAYAAQPFLLRNGANEGFHEAVGEIMSLSAATPKHLKSIGLLSPDFQEDNETEINFLLKQALTIVGTLPFTYMLEKWRWMVFKGEIPKDQWMKKWWEMKREIVGVVEPVPHDETYCDPASLFHVSNDYSFIRYYTRTLYQFQFQEALCQAAKHEGPLHKCDISNSTEAGQKLFNMLRLGKSEPWTLALENVVGAKNMNVRPLLNYFEPLFTWLKDQNKNSFVGWSTDWSPYAD
   ||.|..||.|||.||.|||.|..|||||||.|||||..||||.|..||.|||.|...||.||....|.||.|.|||||||.|||.||.|||||||||||.|||.|||||||||||.|.||||||||||||..||..|.||||||||||.||.||||||||||||||||||||||..||||||||||.|||..|.....|||.|||.|||..|.|||||||.||||||.|||..||..|||.||||||||||||||||||||.|||||||||||||||||..|.|||.||||||||||||.||||||.|||.||||||||...|.||||||||||||||||.|||||||.|||||||||||||||||||.||.||||||||||||||||.||||.|||||||..||||.||.||||||||||.||||.|||||||||||||||||||||||||..||||||||||.|||||||||||||||||||||||.||||||||||||...|||.||||..|||.||||||||||||.||.||..||..|||.|||..|...||.|||.|.|||.|||||.|||..||..|||||.||||||||
 1 STTEDLAKKFLDDFNSEAENLSHQSSLASWEYNTNISDENVQKMDEAGAKWSDFYEKQSKLAKNFSLEEIHNDTVKLQLQILQQSGSPVLSEDKSKRLNSILNAMSTIYSTGKVCKPNNPQECLLLEPGLDNIMGTSKDYNERLWAWEGWRAEVGKQLRPLYEEYVVLKNEMARGYHYEDYGDYWRRDYETEGSPDLEYSRDQLIKDVERIFAEIKPLYEQLHAYVRTKLMDTYPFHISPTGCLPAHLLGDMWGRFWTNLYPLTVPFGQKPNIDVTDAMLNQNWDAKRIFKEAEKFFVSIGLPNMTEGFWNNSMLTDPGDGRKVVCHPTAWDLGKGDFRIKMCTKVTMEDFLTAHHEMGHIQYDMAYASQPYLLRNGANEGFHEAVGEVMSLSVATPKHLKTMGLLSSDFLEDNETEINFLFKQALNIVGTLPFTYMLEKWRWMVFKGEIPKEEWMKKWWEMKRKIVGVVEPVPHDETYCDPASLFHVANDYSFIRYYTRTIFEFQFHEALCRIAKHDGPLHKCDISNSTDAGEKLHQMLSVGKSQPWTSVLKDFVGSKNMDVGPLLRYFEPLYTWLTEQNRKSFVGWNTDWSPYAD
  Score=2643

Normalized score: 4.412353923205342
334 NLCPFGEVFNATRFASVYAWNRKRISNCVADYSVLYNSASFSTFKCYGVSPTKLNDLCFTNVYADSFVIRGDEVRQIAPGQTGKIADYNYKLPDDFTGCVIAWNSNNLDSKVGGNYNYLYRLFRKSNLKPFERDISTEIYQAGSTPCNGVEGFNCYFPLQSYGFQPTNGVGYQPYRVVVLSFELLHAPATVCGP
    ||||||||||||.|.|||||.||.||||||||||||||..||||||||||.||||||||.||||||||..||.||||||||||.||||||||||||.|||.|||..|.|....|||||.||..|...|.|||||||.........||. ....|||.||..|||..|.|.|||||||||||||||.||||||||
  1 NLCPFGEVFNATKFPSVYAWERKKISNCVADYSVLYNSTFFSTFKCYGVSATKLNDLCFSNVYADSFVVKGDDVRQIAPGQTGVIADYNYKLPDDFMGCVLAWNTRNIDATSTGNYNYKYRYLRHGKLRPFERDISNVPFSPDGKPCT-PPALNCYWPLNDYGFYTTTGIGYQPYRVVVLSFELLNAPATVCGP
  Score=792

Normalized score: 4.103626943005182
84 PLQEIQNLTVKLQLQALQQNGSSVLSEDKSKRLN--------------TILNTM--STIYSTGKVCNPDNPQECLLLEPGLNEIMANSLDYNERL--------WAWESWRSEV---------------GKQLRPLYEEYVVLKN----------EMARANHYEDYGDYWRGDYEVNGVDGY
   |..|..|.|......|......|....|.|...|              |.||..  |..|....|...|....   ..||.....|   |||..|        .||.......               || |||....   ..|          .....|.|....||  |.|...|. ||
 4 PFGEVFNATKFPSVYAWERKKISNCVADYSVLYNSTFFSTFKCYGVSATKLNDLCFSNVYADSFVVKGDDVRQ---IAPGQTGVIA---DYNYKLPDDFMGCVLAWNTRNIDATSTGNYNYKYRYLRHGK-LRPFERD---ISNVPFSPDGKPCTPPALNCYWPLNDY--GFYTTTGI-GY
  Score=46.5

Normalized score: 0.24093264248704663
Not valid alignment
Reference summary:
   A -&gt; Q9BYF1
   B -&gt; P0DTC2
Fixed :)
2023-09-15 07:48:46,692 [MainThread  ] [INFO ]  Removed: []
Checking if empty file: /tmp/biobb/unitests_11/fix_pdb/output_fixpdb.pdb
Comparing: 
        File_A: /tmp/biobb/unitests_11/fix_pdb/output_fixpdb.pdb
        File_B: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_fixpdb.pdb
        File_A hash: b&#x27;g\xebw_\xb7\xe0`\x92\xf8\xc0\xbd\xc5\xe6\x86\xbf\xd6\xc9\x8f\x17\x82\xfeV\xd6\xf7\x89v\xf6\x11\x80\x96\xfa_&#x27;
        File_B hash: b&#x27;g\xebw_\xb7\xe0`\x92\xf8\xc0\xbd\xc5\xe6\x86\xbf\xd6\xc9\x8f\x17\x82\xfeV\xd6\xf7\x89v\xf6\x11\x80\x96\xfa_&#x27;
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     /tmp/biobb/unitests_11/fix_pdb/fix_pdb_log.out:file_utils.py:390 Executing biobb_model.model.fix_pdb Version: 4.1.0
INFO     /tmp/biobb/unitests_11/fix_pdb/fix_pdb_log.out:file_utils.py:390 Removed: []<br/> ----------------------------Captured stdout teardown---------------------------- <br/>
Removing: /tmp/biobb/unitests_11
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">biobb_model/test/unitests/test_model/test_fix_side_chain.py::TestFixSideChain::test_launch</td>
          <td class="col-duration">1.34</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>2023-09-15 07:48:46,713 [MainThread  ] [INFO ]  Executing biobb_model.model.fix_side_chain Version: 4.1.0
2023-09-15 07:48:46,714 [MainThread  ] [INFO ]  Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b
2023-09-15 07:48:46,714 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b/2ki5.pdb -o /tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b/output_pdb_path.pdb --force_save fixside --fix ALL

2023-09-15 07:48:47,785 [MainThread  ] [INFO ]  Exit code 0

2023-09-15 07:48:47,785 [MainThread  ] [INFO ]  ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b/2ki5.pdb loaded
 PDB id: 2KI5 
 Title: herpes simplex type-1 thymidine kinase in complex with the drug aciclovir at 1.9a resolution
 Experimental method: x-ray diffraction
 Keywords: transferase (phosphotransferase) thymid, transferase (phosphotransferase) thymidine kinase; viridae; ds-dna enveloped viruses; herpesviridae; alphaherpesvirinae antiviral drug (aciclovir)
 Resolution (A): 1.9

 Num. models: 1
 Num. chains: 2 (A: Protein, B: Protein)
 Num. residues:  908
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  296
 Num. ligands or modified residues:  4
 Num. water mol.:  292
 Num. atoms:  4961
Small mol ligands found
SO4 A4
AC2 A1
SO4 B3
AC2 B2
Running fixside. Options: --fix ALL
Found 6 Residues with missing side chain atoms
 MET A46    CG,SD,CE
 PRO A154   CG,CD
 PRO A280   CG,CD
 MET B46    CG,SD,CE
 PRO B154   CG,CD
 ARG B220   CG,CD,NE,CZ,NH1,NH2
Fixing side chains
MET A46
  Adding new atom CG
  Adding new atom SD
  Adding new atom CE
PRO A154
  Adding new atom CG
  Adding new atom CD
PRO A280
  Adding new atom CG
  Adding new atom CD
MET B46
  Adding new atom CG
  Adding new atom SD
  Adding new atom CE
PRO B154
  Adding new atom CG
  Adding new atom CD
ARG B220
  Adding new atom CG
  Adding new atom CD
  Adding new atom NE
  Adding new atom CZ
  Adding new atom NH1
  Adding new atom NH2
Fixed 6 side chain(s)
Checking for steric clashes
1 Steric severe clashes detected
 MET B46.CE   VAL B352.O      0.921 A
5 Steric apolar clashes detected
 MET A46.CE   VAL A352.CG2    1.843 A
 MET A46.CE   THR A354.O      1.869 A
 MET B46.CE   VAL B352.O      0.921 A
 GLY B61.CA   ARG B220.NH1    2.301 A
 ARG B220.CZ  SO4 B3.O3       1.597 A
3 Steric polar_acceptor clashes detected
 MET A46.SD   VAL A352.O      2.171 A
 MET A46.SD   THR A354.O      2.569 A
 MET B46.SD   VAL B352.O      2.291 A
2 Steric polar_donor clashes detected
 GLY B61.N    ARG B220.NH1    2.443 A
 ARG B216.NH2 ARG B220.NH1    2.977 A
1 Steric positive clashes detected
 ARG B216.NH2 ARG B220.NH1    2.977 A
No negative clashes detected
Final Num. models: 1
Final Num. chains: 2 (A: Protein, B: Protein)
Final Num. residues:  908
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  296
Final Num. ligands or modified residues:  4
Final Num. water mol.:  292
Final Num. atoms:  4979
Small mol ligands found
SO4 A4
AC2 A1
SO4 B3
AC2 B2
Structure saved on /tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b/output_pdb_path.pdb

2023-09-15 07:48:47,786 [MainThread  ] [INFO ]  Removed: [&#x27;/tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b&#x27;]
Checking if empty file: /tmp/biobb/unitests_11/fix_side_chain/output_pdb_path.pdb
Checking RMSD between:
     PDB_A: /tmp/biobb/unitests_11/fix_side_chain/output_pdb_path.pdb
     PDB_B: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_pdb_path.pdb
     Ignoring HETAMT in RMSD
     Ignoring Hydrogen atoms in RMSD
     Atoms ALIGNED in PDB_A: 4937
     Atoms ALIGNED in PDB_B: 4937
     RMS: 6.0339467652777484e-15
     RMS_CUTOFF: 1
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     /tmp/biobb/unitests_11/fix_side_chain/fix_side_chain_log.out:file_utils.py:390 Executing biobb_model.model.fix_side_chain Version: 4.1.0
INFO     /tmp/biobb/unitests_11/fix_side_chain/fix_side_chain_log.out:file_utils.py:390 Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b
INFO     /tmp/biobb/unitests_11/fix_side_chain/fix_side_chain_log.out:cmd_wrapper.py:33 check_structure -i /tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b/2ki5.pdb -o /tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b/output_pdb_path.pdb --force_save fixside --fix ALL

INFO     /tmp/biobb/unitests_11/fix_side_chain/fix_side_chain_log.out:cmd_wrapper.py:50 Exit code 0

INFO     /tmp/biobb/unitests_11/fix_side_chain/fix_side_chain_log.out:cmd_wrapper.py:52 ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b/2ki5.pdb loaded
 PDB id: 2KI5 
 Title: herpes simplex type-1 thymidine kinase in complex with the drug aciclovir at 1.9a resolution
 Experimental method: x-ray diffraction
 Keywords: transferase (phosphotransferase) thymid, transferase (phosphotransferase) thymidine kinase; viridae; ds-dna enveloped viruses; herpesviridae; alphaherpesvirinae antiviral drug (aciclovir)
 Resolution (A): 1.9

 Num. models: 1
 Num. chains: 2 (A: Protein, B: Protein)
 Num. residues:  908
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  296
 Num. ligands or modified residues:  4
 Num. water mol.:  292
 Num. atoms:  4961
Small mol ligands found
SO4 A4
AC2 A1
SO4 B3
AC2 B2
Running fixside. Options: --fix ALL
Found 6 Residues with missing side chain atoms
 MET A46    CG,SD,CE
 PRO A154   CG,CD
 PRO A280   CG,CD
 MET B46    CG,SD,CE
 PRO B154   CG,CD
 ARG B220   CG,CD,NE,CZ,NH1,NH2
Fixing side chains
MET A46
  Adding new atom CG
  Adding new atom SD
  Adding new atom CE
PRO A154
  Adding new atom CG
  Adding new atom CD
PRO A280
  Adding new atom CG
  Adding new atom CD
MET B46
  Adding new atom CG
  Adding new atom SD
  Adding new atom CE
PRO B154
  Adding new atom CG
  Adding new atom CD
ARG B220
  Adding new atom CG
  Adding new atom CD
  Adding new atom NE
  Adding new atom CZ
  Adding new atom NH1
  Adding new atom NH2
Fixed 6 side chain(s)
Checking for steric clashes
1 Steric severe clashes detected
 MET B46.CE   VAL B352.O      0.921 A
5 Steric apolar clashes detected
 MET A46.CE   VAL A352.CG2    1.843 A
 MET A46.CE   THR A354.O      1.869 A
 MET B46.CE   VAL B352.O      0.921 A
 GLY B61.CA   ARG B220.NH1    2.301 A
 ARG B220.CZ  SO4 B3.O3       1.597 A
3 Steric polar_acceptor clashes detected
 MET A46.SD   VAL A352.O      2.171 A
 MET A46.SD   THR A354.O      2.569 A
 MET B46.SD   VAL B352.O      2.291 A
2 Steric polar_donor clashes detected
 GLY B61.N    ARG B220.NH1    2.443 A
 ARG B216.NH2 ARG B220.NH1    2.977 A
1 Steric positive clashes detected
 ARG B216.NH2 ARG B220.NH1    2.977 A
No negative clashes detected
Final Num. models: 1
Final Num. chains: 2 (A: Protein, B: Protein)
Final Num. residues:  908
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  296
Final Num. ligands or modified residues:  4
Final Num. water mol.:  292
Final Num. atoms:  4979
Small mol ligands found
SO4 A4
AC2 A1
SO4 B3
AC2 B2
Structure saved on /tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b/output_pdb_path.pdb

INFO     /tmp/biobb/unitests_11/fix_side_chain/fix_side_chain_log.out:file_utils.py:390 Removed: [&#x27;/tmp/biobb/unitests_11/fix_side_chain/2094539a-db2b-4cd1-9a60-5e10a8e6e27b&#x27;]<br/> ----------------------------Captured stdout teardown---------------------------- <br/>
Removing: /tmp/biobb/unitests_11
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">biobb_model/test/unitests/test_model/test_fix_ssbonds.py::TestFixSSBonds::test_launch</td>
          <td class="col-duration">0.51</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>2023-09-15 07:48:48,059 [MainThread  ] [INFO ]  Executing biobb_model.model.fix_ssbonds Version: 4.1.0
2023-09-15 07:48:48,060 [MainThread  ] [INFO ]  Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/1aki.pdb to /tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f
2023-09-15 07:48:48,060 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f/1aki.pdb -o /tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f/output_pdb_path.pdb --force_save --non_interactive getss --mark All

2023-09-15 07:48:48,505 [MainThread  ] [INFO ]  Exit code 0

2023-09-15 07:48:48,505 [MainThread  ] [INFO ]  ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f/1aki.pdb loaded
 PDB id: 1AKI 
 Title: the structure of the orthorhombic form of hen egg-white lysozyme at 1.5 angstroms resolution
 Experimental method: x-ray diffraction
 Keywords: hydrolase, glycosidase
 Resolution (A): 1.5

 Num. models: 1
 Num. chains: 1 (A: Protein)
 Num. residues:  207
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  78
 Num. ligands or modified residues:  0
 Num. water mol.:  78
 Num. atoms:  1079
Running getss. Options: --mark All
Detected 4 Possible SS Bonds
 CYS A6.SG    CYS A127.SG     3.880
 CYS A30.SG   CYS A115.SG     4.005
 CYS A64.SG   CYS A80.SG      3.948
 CYS A76.SG   CYS A94.SG      4.073
Final Num. models: 1
Final Num. chains: 1 (A: Protein)
Final Num. residues:  207
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  78
Final Num. ligands or modified residues:  0
Final Num. water mol.:  78
Final Num. atoms:  1079
Structure saved on /tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f/output_pdb_path.pdb

2023-09-15 07:48:48,506 [MainThread  ] [INFO ]  Removed: [&#x27;/tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f&#x27;]
Checking if empty file: /tmp/biobb/unitests_11/fix_ssbonds/output_pdb_path.pdb
Checking RMSD between:
     PDB_A: /tmp/biobb/unitests_11/fix_ssbonds/output_pdb_path.pdb
     PDB_B: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_ssbonds.pdb
     Ignoring HETAMT in RMSD
     Ignoring Hydrogen atoms in RMSD
     Atoms ALIGNED in PDB_A: 1079
     Atoms ALIGNED in PDB_B: 1079
     RMS: 1.212094551631224e-14
     RMS_CUTOFF: 1
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     /tmp/biobb/unitests_11/fix_ssbonds/fix_ssbonds_log.out:file_utils.py:390 Executing biobb_model.model.fix_ssbonds Version: 4.1.0
INFO     /tmp/biobb/unitests_11/fix_ssbonds/fix_ssbonds_log.out:file_utils.py:390 Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/1aki.pdb to /tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f
INFO     /tmp/biobb/unitests_11/fix_ssbonds/fix_ssbonds_log.out:cmd_wrapper.py:33 check_structure -i /tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f/1aki.pdb -o /tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f/output_pdb_path.pdb --force_save --non_interactive getss --mark All

INFO     /tmp/biobb/unitests_11/fix_ssbonds/fix_ssbonds_log.out:cmd_wrapper.py:50 Exit code 0

INFO     /tmp/biobb/unitests_11/fix_ssbonds/fix_ssbonds_log.out:cmd_wrapper.py:52 ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f/1aki.pdb loaded
 PDB id: 1AKI 
 Title: the structure of the orthorhombic form of hen egg-white lysozyme at 1.5 angstroms resolution
 Experimental method: x-ray diffraction
 Keywords: hydrolase, glycosidase
 Resolution (A): 1.5

 Num. models: 1
 Num. chains: 1 (A: Protein)
 Num. residues:  207
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  78
 Num. ligands or modified residues:  0
 Num. water mol.:  78
 Num. atoms:  1079
Running getss. Options: --mark All
Detected 4 Possible SS Bonds
 CYS A6.SG    CYS A127.SG     3.880
 CYS A30.SG   CYS A115.SG     4.005
 CYS A64.SG   CYS A80.SG      3.948
 CYS A76.SG   CYS A94.SG      4.073
Final Num. models: 1
Final Num. chains: 1 (A: Protein)
Final Num. residues:  207
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  78
Final Num. ligands or modified residues:  0
Final Num. water mol.:  78
Final Num. atoms:  1079
Structure saved on /tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f/output_pdb_path.pdb

INFO     /tmp/biobb/unitests_11/fix_ssbonds/fix_ssbonds_log.out:file_utils.py:390 Removed: [&#x27;/tmp/biobb/unitests_11/fix_ssbonds/dabcfe28-daaf-4e60-8e49-edb26855447f&#x27;]<br/> ----------------------------Captured stdout teardown---------------------------- <br/>
Removing: /tmp/biobb/unitests_11
<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">biobb_model/test/unitests/test_model/test_mutate.py::TestFixSideChain::test_launch</td>
          <td class="col-duration">1.18</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log"> ------------------------------Captured stdout call------------------------------ <br/>2023-09-15 07:48:48,574 [MainThread  ] [INFO ]  Executing biobb_model.model.mutate Version: 4.1.0
2023-09-15 07:48:48,575 [MainThread  ] [INFO ]  Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd
2023-09-15 07:48:48,575 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd/2ki5.pdb -o /tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd/output_mutated_pdb_path.pdb --force_save --non_interactive mutateside --mut Leu49Ile,B:arg51Lys

2023-09-15 07:48:49,501 [MainThread  ] [INFO ]  Exit code 0

2023-09-15 07:48:49,501 [MainThread  ] [INFO ]  ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd/2ki5.pdb loaded
 PDB id: 2KI5 
 Title: herpes simplex type-1 thymidine kinase in complex with the drug aciclovir at 1.9a resolution
 Experimental method: x-ray diffraction
 Keywords: transferase (phosphotransferase) thymid, transferase (phosphotransferase) thymidine kinase; viridae; ds-dna enveloped viruses; herpesviridae; alphaherpesvirinae antiviral drug (aciclovir)
 Resolution (A): 1.9

 Num. models: 1
 Num. chains: 2 (A: Protein, B: Protein)
 Num. residues:  908
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  296
 Num. ligands or modified residues:  4
 Num. water mol.:  292
 Num. atoms:  4961
Small mol ligands found
SO4 A4
AC2 A1
SO4 B3
AC2 B2
Running mutateside. Options: --mut Leu49Ile,B:arg51Lys
Mutations to perform
*:LEU49ILE
B:ARG51LYS
Replacing LEU A49 into ILE
  Deleting CD2
  Renaming CG to CG1
  Adding new atom CG2
Replacing LEU B49 into ILE
  Deleting CD2
  Renaming CG to CG1
  Adding new atom CG2

Replacing ARG B51 into LYS
  Deleting NE
  Deleting CZ
  Deleting NH1
  Deleting NH2
  Adding new atom CE
  Adding new atom NZ

Checking for steric clashes
No severe clashes detected
4 Steric apolar clashes detected
 ILE A49.CG2  THR A350.OG1    2.815 A
 ILE B49.CG2  THR B350.OG1    2.611 A
 LYS B51.NZ   LEU B139.CG     2.403 A
 LYS B51.NZ   ILE B143.CD1    2.319 A
No polar_acceptor clashes detected
No polar_donor clashes detected
No positive clashes detected
No negative clashes detected
Final Num. models: 1
Final Num. chains: 2 (A: Protein, B: Protein)
Final Num. residues:  908
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  296
Final Num. ligands or modified residues:  4
Final Num. water mol.:  292
Final Num. atoms:  4959
Small mol ligands found
SO4 A4
AC2 A1
SO4 B3
AC2 B2
Structure saved on /tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd/output_mutated_pdb_path.pdb

2023-09-15 07:48:49,502 [MainThread  ] [INFO ]  Removed: [&#x27;/tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd&#x27;]
Checking if empty file: /tmp/biobb/unitests_11/mutate/output_mutated_pdb_path.pdb
Checking RMSD between:
     PDB_A: /tmp/biobb/unitests_11/mutate/output_mutated_pdb_path.pdb
     PDB_B: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_mutated_pdb_path.pdb
     Ignoring HETAMT in RMSD
     Ignoring Hydrogen atoms in RMSD
     Atoms ALIGNED in PDB_A: 4917
     Atoms ALIGNED in PDB_B: 4917
     RMS: 1.983345332028294e-14
     RMS_CUTOFF: 1
<br/> -------------------------------Captured log call-------------------------------- <br/>INFO     /tmp/biobb/unitests_11/mutate/mutate_log.out:file_utils.py:390 Executing biobb_model.model.mutate Version: 4.1.0
INFO     /tmp/biobb/unitests_11/mutate/mutate_log.out:file_utils.py:390 Copy: /home/user/actions-runner/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd
INFO     /tmp/biobb/unitests_11/mutate/mutate_log.out:cmd_wrapper.py:33 check_structure -i /tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd/2ki5.pdb -o /tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd/output_mutated_pdb_path.pdb --force_save --non_interactive mutateside --mut Leu49Ile,B:arg51Lys

INFO     /tmp/biobb/unitests_11/mutate/mutate_log.out:cmd_wrapper.py:50 Exit code 0

INFO     /tmp/biobb/unitests_11/mutate/mutate_log.out:cmd_wrapper.py:52 ================================================================================
=                   BioBB structure checking utility v3.13.4                   =
=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =
================================================================================

Warning: sequence features may not be available, use --sequence for  external fasta input
Structure /tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd/2ki5.pdb loaded
 PDB id: 2KI5 
 Title: herpes simplex type-1 thymidine kinase in complex with the drug aciclovir at 1.9a resolution
 Experimental method: x-ray diffraction
 Keywords: transferase (phosphotransferase) thymid, transferase (phosphotransferase) thymidine kinase; viridae; ds-dna enveloped viruses; herpesviridae; alphaherpesvirinae antiviral drug (aciclovir)
 Resolution (A): 1.9

 Num. models: 1
 Num. chains: 2 (A: Protein, B: Protein)
 Num. residues:  908
 Num. residues with ins. codes:  0
 Num. residues with H atoms: 0
 Num. HETATM residues:  296
 Num. ligands or modified residues:  4
 Num. water mol.:  292
 Num. atoms:  4961
Small mol ligands found
SO4 A4
AC2 A1
SO4 B3
AC2 B2
Running mutateside. Options: --mut Leu49Ile,B:arg51Lys
Mutations to perform
*:LEU49ILE
B:ARG51LYS
Replacing LEU A49 into ILE
  Deleting CD2
  Renaming CG to CG1
  Adding new atom CG2
Replacing LEU B49 into ILE
  Deleting CD2
  Renaming CG to CG1
  Adding new atom CG2

Replacing ARG B51 into LYS
  Deleting NE
  Deleting CZ
  Deleting NH1
  Deleting NH2
  Adding new atom CE
  Adding new atom NZ

Checking for steric clashes
No severe clashes detected
4 Steric apolar clashes detected
 ILE A49.CG2  THR A350.OG1    2.815 A
 ILE B49.CG2  THR B350.OG1    2.611 A
 LYS B51.NZ   LEU B139.CG     2.403 A
 LYS B51.NZ   ILE B143.CD1    2.319 A
No polar_acceptor clashes detected
No polar_donor clashes detected
No positive clashes detected
No negative clashes detected
Final Num. models: 1
Final Num. chains: 2 (A: Protein, B: Protein)
Final Num. residues:  908
Final Num. residues with ins. codes:  0
Final Num. residues with H atoms: 0
Final Num. HETATM residues:  296
Final Num. ligands or modified residues:  4
Final Num. water mol.:  292
Final Num. atoms:  4959
Small mol ligands found
SO4 A4
AC2 A1
SO4 B3
AC2 B2
Structure saved on /tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd/output_mutated_pdb_path.pdb

INFO     /tmp/biobb/unitests_11/mutate/mutate_log.out:file_utils.py:390 Removed: [&#x27;/tmp/biobb/unitests_11/mutate/5d3359f5-feb1-45c6-8e0f-06bd2ce6b9bd&#x27;]<br/> ----------------------------Captured stdout teardown---------------------------- <br/>
Removing: /tmp/biobb/unitests_11
<br/></div></td></tr></tbody></table></body></html>