<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <title id="head-title">report.html</title>
      <link href="assets/style.css" rel="stylesheet" type="text/css"/>
  </head>
  <body>
    <h1 id="title">report.html</h1>
    <p>Report generated on 06-Jun-2025 at 12:44:07 by <a href="https://pypi.python.org/pypi/pytest-html">pytest-html</a>
        v4.1.1</p>
    <div id="environment-header">
      <h2>Environment</h2>
    </div>
    <table id="environment"></table>
    <!-- TEMPLATES -->
      <template id="template_environment_row">
      <tr>
        <td></td>
        <td></td>
      </tr>
    </template>
    <template id="template_results-table__body--empty">
      <tbody class="results-table-row">
        <tr id="not-found-message">
          <td colspan="4">No results found. Check the filters.</th>
        </tr>
    </template>
    <template id="template_results-table__tbody">
      <tbody class="results-table-row">
        <tr class="collapsible">
        </tr>
        <tr class="extras-row">
          <td class="extra" colspan="4">
            <div class="extraHTML"></div>
            <div class="media">
              <div class="media-container">
                  <div class="media-container__nav--left"><</div>
                  <div class="media-container__viewport">
                    <img src="" />
                    <video controls>
                      <source src="" type="video/mp4">
                    </video>
                  </div>
                  <div class="media-container__nav--right">></div>
                </div>
                <div class="media__name"></div>
                <div class="media__counter"></div>
            </div>
            <div class="logwrapper">
              <div class="logexpander"></div>
              <div class="log"></div>
            </div>
          </td>
        </tr>
      </tbody>
    </template>
    <!-- END TEMPLATES -->
    <div class="summary">
      <div class="summary__data">
        <h2>Summary</h2>
        <div class="additional-summary prefix">
        </div>
        <p class="run-count">8 tests took 00:01:04.</p>
        <p class="filter">(Un)check the boxes to filter the results.</p>
        <div class="summary__reload">
          <div class="summary__reload__button hidden" onclick="location.reload()">
            <div>There are still tests running. <br />Reload this page to get the latest results!</div>
          </div>
        </div>
        <div class="summary__spacer"></div>
        <div class="controls">
          <div class="filters">
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="failed" disabled/>
            <span class="failed">0 Failed,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="passed" />
            <span class="passed">8 Passed,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="skipped" disabled/>
            <span class="skipped">0 Skipped,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="xfailed" disabled/>
            <span class="xfailed">0 Expected failures,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="xpassed" disabled/>
            <span class="xpassed">0 Unexpected passes,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="error" disabled/>
            <span class="error">0 Errors,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="rerun" disabled/>
            <span class="rerun">0 Reruns</span>
          </div>
          <div class="collapse">
            <button id="show_all_details">Show all details</button>&nbsp;/&nbsp;<button id="hide_all_details">Hide all details</button>
          </div>
        </div>
      </div>
      <div class="additional-summary summary">
      </div>
      <div class="additional-summary postfix">
      </div>
    </div>
    <table id="results-table">
      <thead id="results-table-head">
        <tr>
          <th class="sortable" data-column-type="result">Result</th>
          <th class="sortable" data-column-type="testId">Test</th>
          <th class="sortable" data-column-type="duration">Duration</th>
          <th>Links</th>
        </tr>
      </thead>
    </table>
  </body>
  <footer>
    <div id="data-container" data-jsonblob="{&#34;environment&#34;: {&#34;Python&#34;: &#34;3.12.11&#34;, &#34;Platform&#34;: &#34;Linux-5.15.0-125-generic-x86_64-with-glibc2.35&#34;, &#34;Packages&#34;: {&#34;pytest&#34;: &#34;8.4.0&#34;, &#34;pluggy&#34;: &#34;1.6.0&#34;}, &#34;Plugins&#34;: {&#34;cov&#34;: &#34;6.1.1&#34;, &#34;html&#34;: &#34;4.1.1&#34;, &#34;metadata&#34;: &#34;3.1.1&#34;}, &#34;CI&#34;: &#34;true&#34;}, &#34;tests&#34;: {&#34;biobb_model/test/unitests/test_model/test_checking_log.py::TestCheckingLog::test_launch&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;biobb_model/test/unitests/test_model/test_checking_log.py::TestCheckingLog::test_launch&#34;, &#34;duration&#34;: &#34;00:00:35&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;biobb_model/test/unitests/test_model/test_checking_log.py::TestCheckingLog::test_launch&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:35&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;----------------------------- Captured stdout call -----------------------------\n2025-06-06 12:43:02,725 [MainThread  ] [INFO ]  Module: biobb_model.model.checking_log Version: 5.0.0\n2025-06-06 12:43:02,726 [MainThread  ] [INFO ]  /tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49f2-8aff-e474bd05ae74 directory successfully created\n2025-06-06 12:43:02,728 [MainThread  ] [INFO ]  Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49f2-8aff-e474bd05ae74\n2025-06-06 12:43:02,729 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49f2-8aff-e474bd05ae74/2ki5.pdb checkall &amp;gt; /tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49f2-8aff-e474bd05ae74/output_log_path.log\n\n2025-06-06 12:43:37,683 [MainThread  ] [INFO ]  Executing: check_structure -i /tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49...\n2025-06-06 12:43:37,683 [MainThread  ] [INFO ]  Exit code: 0\n2025-06-06 12:43:37,686 [MainThread  ] [INFO ]  Removed: [&amp;#x27;/tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49f2-8aff-e474bd05ae74&amp;#x27;]\n2025-06-06 12:43:37,687 [MainThread  ] [INFO ]  \nChecking if empty file: /tmp/biobb/unitests_120/checking_log/output_log_path.log\n\n------------------------------ Captured log call -------------------------------\nINFO     /tmp/biobb/unitests_120/checking_log/checking_log_log.out:file_utils.py:409 Module: biobb_model.model.checking_log Version: 5.0.0\nINFO     /tmp/biobb/unitests_120/checking_log/checking_log_log.out:file_utils.py:82 /tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49f2-8aff-e474bd05ae74 directory successfully created\nINFO     /tmp/biobb/unitests_120/checking_log/checking_log_log.out:file_utils.py:409 Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49f2-8aff-e474bd05ae74\nINFO     /tmp/biobb/unitests_120/checking_log/checking_log_log.out:cmd_wrapper.py:63 check_structure -i /tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49f2-8aff-e474bd05ae74/2ki5.pdb checkall &amp;gt; /tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49f2-8aff-e474bd05ae74/output_log_path.log\n\nINFO     /tmp/biobb/unitests_120/checking_log/checking_log_log.out:cmd_wrapper.py:36 Executing: check_structure -i /tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49...\nINFO     /tmp/biobb/unitests_120/checking_log/checking_log_log.out:cmd_wrapper.py:37 Exit code: 0\nINFO     /tmp/biobb/unitests_120/checking_log/checking_log_log.out:file_utils.py:409 Removed: [&amp;#x27;/tmp/biobb/unitests_120/checking_log/sandbox_7cb73ee3-4328-49f2-8aff-e474bd05ae74&amp;#x27;]\nINFO     /tmp/biobb/unitests_120/checking_log/checking_log_log.out:file_utils.py:409\n\n--------------------------- Captured stdout teardown ---------------------------\n\nRemoving: /tmp/biobb/unitests_120\n&#34;}], &#34;biobb_model/test/unitests/test_model/test_fix_altlocs.py::TestFixAltlocs::test_launch&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;biobb_model/test/unitests/test_model/test_fix_altlocs.py::TestFixAltlocs::test_launch&#34;, &#34;duration&#34;: &#34;00:00:03&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;biobb_model/test/unitests/test_model/test_fix_altlocs.py::TestFixAltlocs::test_launch&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:03&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;----------------------------- Captured stdout call -----------------------------\n2025-06-06 12:43:37,761 [MainThread  ] [INFO ]  Module: biobb_model.model.fix_altlocs Version: 5.0.0\n2025-06-06 12:43:37,762 [MainThread  ] [INFO ]  /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947 directory successfully created\n2025-06-06 12:43:37,764 [MainThread  ] [INFO ]  Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/3ebp.pdb to /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947\n2025-06-06 12:43:37,765 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947/3ebp.pdb -o /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947/output_pdb_path.pdb --force_save --non_interactive altloc --select A339:A,A171:B,A768:A\n\n2025-06-06 12:43:40,257 [MainThread  ] [INFO ]  Executing: check_structure -i /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-494...\n2025-06-06 12:43:40,257 [MainThread  ] [INFO ]  Exit code: 0\n2025-06-06 12:43:40,257 [MainThread  ] [INFO ]  ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947/3ebp.pdb loaded\n PDB id: 3EBP \n Title: glycogen phosphorylase b/flavopiridol complex\n Experimental method: x-ray diffraction\n Keywords: glycogen phosphorylase, rational inhibitor design, glycogenolysis, diabetes type 2, allosteric enzyme, carbohydrate metabolism, glycogen metabolism, glycosyltransferase, nucleotide-binding, phosphoprotein, pyridoxal phosphate, transferase-transferase inhibitor complex\n Resolution (A): 2.0\n\n Num. models: 1\n Num. chains: 1 (A: Protein)\n Num. residues:  1097\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  288\n Num. ligands or modified residues:  2\n Num. water mol.:  286\n Num. atoms:  6918\nModified residues found\nLLP A680\nSmall mol ligands found\nCPB A940\nRunning altloc. Options: --select A339:A,A171:B,A768:A\nDetected 3 residues with alternative location labels\nCYS A171\n  CB   A (0.50) B (0.50)\n  SG   A (0.50) B (0.50)\nASP A339\n  CG   A (0.50) B (0.50)\n  OD1  A (0.50) B (0.50)\n  OD2  A (0.50) B (0.50)\nHIS A768\n  CG   A (0.50) B (0.50)\n  ND1  A (0.50) B (0.50)\n  CD2  A (0.50) B (0.50)\n  CE1  A (0.50) B (0.50)\n  NE2  A (0.50) B (0.50)\nSelecting location A339:A,A171:B,A768:A\nFinal Num. models: 1\nFinal Num. chains: 1 (A: Protein)\nFinal Num. residues:  1097\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  288\nFinal Num. ligands or modified residues:  2\nFinal Num. water mol.:  286\nFinal Num. atoms:  6918\nModified residues found\nLLP A680\nSmall mol ligands found\nCPB A940\nStructure saved on /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947/output_pdb_path.pdb\n\n2025-06-06 12:43:40,260 [MainThread  ] [INFO ]  Removed: [&amp;#x27;/tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947&amp;#x27;]\n2025-06-06 12:43:40,261 [MainThread  ] [INFO ]  \nChecking if empty file: /tmp/biobb/unitests_120/fix_altlocs/output_pdb_path.pdb\nChecking RMSD between:\n     PDB_A: /tmp/biobb/unitests_120/fix_altlocs/output_pdb_path.pdb\n     PDB_B: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_altloc.pdb\n     Ignoring HETAMT in RMSD\n     Ignoring Hydrogen atoms in RMSD\n     Atoms ALIGNED in PDB_A: 6866\n     Atoms ALIGNED in PDB_B: 6866\n     RMS: 1.066577713794962e-14\n     RMS_CUTOFF: 1\n\n------------------------------ Captured log call -------------------------------\nINFO     /tmp/biobb/unitests_120/fix_altlocs/fix_altlocs_log.out:file_utils.py:409 Module: biobb_model.model.fix_altlocs Version: 5.0.0\nINFO     /tmp/biobb/unitests_120/fix_altlocs/fix_altlocs_log.out:file_utils.py:82 /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947 directory successfully created\nINFO     /tmp/biobb/unitests_120/fix_altlocs/fix_altlocs_log.out:file_utils.py:409 Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/3ebp.pdb to /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947\nINFO     /tmp/biobb/unitests_120/fix_altlocs/fix_altlocs_log.out:cmd_wrapper.py:63 check_structure -i /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947/3ebp.pdb -o /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947/output_pdb_path.pdb --force_save --non_interactive altloc --select A339:A,A171:B,A768:A\n\nINFO     /tmp/biobb/unitests_120/fix_altlocs/fix_altlocs_log.out:cmd_wrapper.py:36 Executing: check_structure -i /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-494...\nINFO     /tmp/biobb/unitests_120/fix_altlocs/fix_altlocs_log.out:cmd_wrapper.py:37 Exit code: 0\nINFO     /tmp/biobb/unitests_120/fix_altlocs/fix_altlocs_log.out:cmd_wrapper.py:41 ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947/3ebp.pdb loaded\n PDB id: 3EBP \n Title: glycogen phosphorylase b/flavopiridol complex\n Experimental method: x-ray diffraction\n Keywords: glycogen phosphorylase, rational inhibitor design, glycogenolysis, diabetes type 2, allosteric enzyme, carbohydrate metabolism, glycogen metabolism, glycosyltransferase, nucleotide-binding, phosphoprotein, pyridoxal phosphate, transferase-transferase inhibitor complex\n Resolution (A): 2.0\n\n Num. models: 1\n Num. chains: 1 (A: Protein)\n Num. residues:  1097\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  288\n Num. ligands or modified residues:  2\n Num. water mol.:  286\n Num. atoms:  6918\nModified residues found\nLLP A680\nSmall mol ligands found\nCPB A940\nRunning altloc. Options: --select A339:A,A171:B,A768:A\nDetected 3 residues with alternative location labels\nCYS A171\n  CB   A (0.50) B (0.50)\n  SG   A (0.50) B (0.50)\nASP A339\n  CG   A (0.50) B (0.50)\n  OD1  A (0.50) B (0.50)\n  OD2  A (0.50) B (0.50)\nHIS A768\n  CG   A (0.50) B (0.50)\n  ND1  A (0.50) B (0.50)\n  CD2  A (0.50) B (0.50)\n  CE1  A (0.50) B (0.50)\n  NE2  A (0.50) B (0.50)\nSelecting location A339:A,A171:B,A768:A\nFinal Num. models: 1\nFinal Num. chains: 1 (A: Protein)\nFinal Num. residues:  1097\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  288\nFinal Num. ligands or modified residues:  2\nFinal Num. water mol.:  286\nFinal Num. atoms:  6918\nModified residues found\nLLP A680\nSmall mol ligands found\nCPB A940\nStructure saved on /tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947/output_pdb_path.pdb\n\nINFO     /tmp/biobb/unitests_120/fix_altlocs/fix_altlocs_log.out:file_utils.py:409 Removed: [&amp;#x27;/tmp/biobb/unitests_120/fix_altlocs/sandbox_076a2c86-be89-4945-89e8-6c39a2282947&amp;#x27;]\nINFO     /tmp/biobb/unitests_120/fix_altlocs/fix_altlocs_log.out:file_utils.py:409\n\n--------------------------- Captured stdout teardown ---------------------------\n\nRemoving: /tmp/biobb/unitests_120\n&#34;}], &#34;biobb_model/test/unitests/test_model/test_fix_amides.py::TestFixAmides::test_launch&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;biobb_model/test/unitests/test_model/test_fix_amides.py::TestFixAmides::test_launch&#34;, &#34;duration&#34;: &#34;00:00:06&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;biobb_model/test/unitests/test_model/test_fix_amides.py::TestFixAmides::test_launch&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:06&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;----------------------------- Captured stdout call -----------------------------\n2025-06-06 12:43:41,121 [MainThread  ] [INFO ]  Module: biobb_model.model.fix_amides Version: 5.0.0\n2025-06-06 12:43:41,122 [MainThread  ] [INFO ]  /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279 directory successfully created\n2025-06-06 12:43:41,124 [MainThread  ] [INFO ]  Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/5s2z.pdb to /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279\n2025-06-06 12:43:41,125 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279/5s2z.pdb -o /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279/output_pdb_path.pdb --force_save amide --fix All\n\n2025-06-06 12:43:46,966 [MainThread  ] [INFO ]  Executing: check_structure -i /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f...\n2025-06-06 12:43:46,967 [MainThread  ] [INFO ]  Exit code: 0\n2025-06-06 12:43:46,967 [MainThread  ] [INFO ]  ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279/5s2z.pdb loaded\n PDB id: 5S2Z \n Title: pandda analysis group deposition -- crystal structure of sars-cov-2 nsp3 macrodomain in complex with z126932614\n Experimental method: x-ray diffraction\n Keywords: sgc - diamond i04-1 fragment screening, pandda, xchemexplorer, sars- cov-2 nsp3 macrodomain, viral protein\n Resolution (A): 1.07\n\n Num. models: 1\n Num. chains: 2 (A: Protein, B: Protein)\n Num. residues:  792\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  458\n Num. ligands or modified residues:  1\n Num. water mol.:  457\n Num. atoms:  2998\nModified residues found\nT6J A401\nRunning amide. Options: --fix All\nFound 2 unusual contact(s) involving amide atoms\n ASN B20.OD1  VAL B155.O      3.015 A\n ASN B20.OD1  PHE B156.O      2.894 A\nAmide residues fixed All (1)\nRechecking\nFound no unusual contact(s) involving amide atoms\nFinal Num. models: 1\nFinal Num. chains: 2 (A: Protein, B: Protein)\nFinal Num. residues:  792\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  458\nFinal Num. ligands or modified residues:  1\nFinal Num. water mol.:  457\nFinal Num. atoms:  2998\nModified residues found\nT6J A401\nStructure saved on /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279/output_pdb_path.pdb\n\n2025-06-06 12:43:46,969 [MainThread  ] [INFO ]  Removed: [&amp;#x27;/tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279&amp;#x27;]\n2025-06-06 12:43:46,970 [MainThread  ] [INFO ]  \nChecking if empty file: /tmp/biobb/unitests_120/fix_amides/output_pdb_path.pdb\nChecking RMSD between:\n     PDB_A: /tmp/biobb/unitests_120/fix_amides/output_pdb_path.pdb\n     PDB_B: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_amide_pdb_path.pdb\n     Ignoring HETAMT in RMSD\n     Ignoring Hydrogen atoms in RMSD\n     Atoms ALIGNED in PDB_A: 2984\n     Atoms ALIGNED in PDB_B: 2984\n     RMS: 8.13974682237932e-15\n     RMS_CUTOFF: 1\n\n------------------------------ Captured log call -------------------------------\nINFO     /tmp/biobb/unitests_120/fix_amides/fix_amides_log.out:file_utils.py:409 Module: biobb_model.model.fix_amides Version: 5.0.0\nINFO     /tmp/biobb/unitests_120/fix_amides/fix_amides_log.out:file_utils.py:82 /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279 directory successfully created\nINFO     /tmp/biobb/unitests_120/fix_amides/fix_amides_log.out:file_utils.py:409 Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/5s2z.pdb to /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279\nINFO     /tmp/biobb/unitests_120/fix_amides/fix_amides_log.out:cmd_wrapper.py:63 check_structure -i /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279/5s2z.pdb -o /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279/output_pdb_path.pdb --force_save amide --fix All\n\nINFO     /tmp/biobb/unitests_120/fix_amides/fix_amides_log.out:cmd_wrapper.py:36 Executing: check_structure -i /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f...\nINFO     /tmp/biobb/unitests_120/fix_amides/fix_amides_log.out:cmd_wrapper.py:37 Exit code: 0\nINFO     /tmp/biobb/unitests_120/fix_amides/fix_amides_log.out:cmd_wrapper.py:41 ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279/5s2z.pdb loaded\n PDB id: 5S2Z \n Title: pandda analysis group deposition -- crystal structure of sars-cov-2 nsp3 macrodomain in complex with z126932614\n Experimental method: x-ray diffraction\n Keywords: sgc - diamond i04-1 fragment screening, pandda, xchemexplorer, sars- cov-2 nsp3 macrodomain, viral protein\n Resolution (A): 1.07\n\n Num. models: 1\n Num. chains: 2 (A: Protein, B: Protein)\n Num. residues:  792\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  458\n Num. ligands or modified residues:  1\n Num. water mol.:  457\n Num. atoms:  2998\nModified residues found\nT6J A401\nRunning amide. Options: --fix All\nFound 2 unusual contact(s) involving amide atoms\n ASN B20.OD1  VAL B155.O      3.015 A\n ASN B20.OD1  PHE B156.O      2.894 A\nAmide residues fixed All (1)\nRechecking\nFound no unusual contact(s) involving amide atoms\nFinal Num. models: 1\nFinal Num. chains: 2 (A: Protein, B: Protein)\nFinal Num. residues:  792\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  458\nFinal Num. ligands or modified residues:  1\nFinal Num. water mol.:  457\nFinal Num. atoms:  2998\nModified residues found\nT6J A401\nStructure saved on /tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279/output_pdb_path.pdb\n\nINFO     /tmp/biobb/unitests_120/fix_amides/fix_amides_log.out:file_utils.py:409 Removed: [&amp;#x27;/tmp/biobb/unitests_120/fix_amides/sandbox_7b76f78e-6923-4f0f-b3c7-7354c59e0279&amp;#x27;]\nINFO     /tmp/biobb/unitests_120/fix_amides/fix_amides_log.out:file_utils.py:409\n\n--------------------------- Captured stdout teardown ---------------------------\n\nRemoving: /tmp/biobb/unitests_120\n&#34;}], &#34;biobb_model/test/unitests/test_model/test_fix_chirality.py::TestFixChirality::test_launch&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;biobb_model/test/unitests/test_model/test_fix_chirality.py::TestFixChirality::test_launch&#34;, &#34;duration&#34;: &#34;00:00:04&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;biobb_model/test/unitests/test_model/test_fix_chirality.py::TestFixChirality::test_launch&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:04&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;----------------------------- Captured stdout call -----------------------------\n2025-06-06 12:43:47,362 [MainThread  ] [INFO ]  Module: biobb_model.model.fix_chirality Version: 5.0.0\n2025-06-06 12:43:47,363 [MainThread  ] [INFO ]  /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56 directory successfully created\n2025-06-06 12:43:47,365 [MainThread  ] [INFO ]  Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/3e2y.pdb to /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56\n2025-06-06 12:43:47,366 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56/3e2y.pdb -o /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56/output_pdb_path.pdb --force_save chiral --fix All\n\n2025-06-06 12:43:50,459 [MainThread  ] [INFO ]  Executing: check_structure -i /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-4...\n2025-06-06 12:43:50,459 [MainThread  ] [INFO ]  Exit code: 0\n2025-06-06 12:43:50,460 [MainThread  ] [INFO ]  ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56/3e2y.pdb loaded\n PDB id: 3E2Y \n Title: crystal structure of mouse kynurenine aminotransferase iii in complex with glutamine\n Experimental method: x-ray diffraction\n Keywords: alpha beta protein, plp dependent protein, aminotransferase, lyase, pyridoxal phosphate, transferase\n Resolution (A): 2.26\n\n Num. models: 1\n Num. chains: 2 (A: Protein, B: Protein)\n Num. residues:  1239\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  419\n Num. ligands or modified residues:  8\n Num. water mol.:  411\n Num. atoms:  6993\nModified residues found\nGLN A456\nGLN B457\nSmall mol ligands found\nPMP A457\nGOL A458\nGOL A459\nPMP B456\nGOL B458\nGOL B459\nRunning chiral. Options: --fix All\nFound 2 residues with incorrect side-chain chirality\n THR A256  \n THR B286  \nChiral side chains fixed All (2)\nChecking for steric clashes\nNo Severe Steric clashes/covalent bonds detected\n1 Apolar steric clashes detected\n THR A256.CG2 GLU A348.OE2    2.678 A\nNo Polar contacts (acceptors) detected\n1 Polar contacts (donors) detected\n GLN A97.NE2  THR B286.N      3.001 A\nNo Positively charged contacts detected\nNo Negatively charged contacts detected\nFinal Num. models: 1\nFinal Num. chains: 2 (A: Protein, B: Protein)\nFinal Num. residues:  1239\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  419\nFinal Num. ligands or modified residues:  8\nFinal Num. water mol.:  411\nFinal Num. atoms:  6993\nModified residues found\nGLN A456\nGLN B457\nSmall mol ligands found\nPMP A457\nGOL A458\nGOL A459\nPMP B456\nGOL B458\nGOL B459\nStructure saved on /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56/output_pdb_path.pdb\n\n2025-06-06 12:43:50,463 [MainThread  ] [INFO ]  Removed: [&amp;#x27;/tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56&amp;#x27;]\n2025-06-06 12:43:50,463 [MainThread  ] [INFO ]  \nChecking if empty file: /tmp/biobb/unitests_120/fix_chirality/output_pdb_path.pdb\nChecking RMSD between:\n     PDB_A: /tmp/biobb/unitests_120/fix_chirality/output_pdb_path.pdb\n     PDB_B: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_chirality_pdb_path.pdb\n     Ignoring HETAMT in RMSD\n     Ignoring Hydrogen atoms in RMSD\n     Atoms ALIGNED in PDB_A: 6917\n     Atoms ALIGNED in PDB_B: 6917\n     RMS: 1.1372082567343243e-14\n     RMS_CUTOFF: 1\n\n------------------------------ Captured log call -------------------------------\nINFO     /tmp/biobb/unitests_120/fix_chirality/fix_chirality_log.out:file_utils.py:409 Module: biobb_model.model.fix_chirality Version: 5.0.0\nINFO     /tmp/biobb/unitests_120/fix_chirality/fix_chirality_log.out:file_utils.py:82 /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56 directory successfully created\nINFO     /tmp/biobb/unitests_120/fix_chirality/fix_chirality_log.out:file_utils.py:409 Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/3e2y.pdb to /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56\nINFO     /tmp/biobb/unitests_120/fix_chirality/fix_chirality_log.out:cmd_wrapper.py:63 check_structure -i /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56/3e2y.pdb -o /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56/output_pdb_path.pdb --force_save chiral --fix All\n\nINFO     /tmp/biobb/unitests_120/fix_chirality/fix_chirality_log.out:cmd_wrapper.py:36 Executing: check_structure -i /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-4...\nINFO     /tmp/biobb/unitests_120/fix_chirality/fix_chirality_log.out:cmd_wrapper.py:37 Exit code: 0\nINFO     /tmp/biobb/unitests_120/fix_chirality/fix_chirality_log.out:cmd_wrapper.py:41 ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56/3e2y.pdb loaded\n PDB id: 3E2Y \n Title: crystal structure of mouse kynurenine aminotransferase iii in complex with glutamine\n Experimental method: x-ray diffraction\n Keywords: alpha beta protein, plp dependent protein, aminotransferase, lyase, pyridoxal phosphate, transferase\n Resolution (A): 2.26\n\n Num. models: 1\n Num. chains: 2 (A: Protein, B: Protein)\n Num. residues:  1239\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  419\n Num. ligands or modified residues:  8\n Num. water mol.:  411\n Num. atoms:  6993\nModified residues found\nGLN A456\nGLN B457\nSmall mol ligands found\nPMP A457\nGOL A458\nGOL A459\nPMP B456\nGOL B458\nGOL B459\nRunning chiral. Options: --fix All\nFound 2 residues with incorrect side-chain chirality\n THR A256  \n THR B286  \nChiral side chains fixed All (2)\nChecking for steric clashes\nNo Severe Steric clashes/covalent bonds detected\n1 Apolar steric clashes detected\n THR A256.CG2 GLU A348.OE2    2.678 A\nNo Polar contacts (acceptors) detected\n1 Polar contacts (donors) detected\n GLN A97.NE2  THR B286.N      3.001 A\nNo Positively charged contacts detected\nNo Negatively charged contacts detected\nFinal Num. models: 1\nFinal Num. chains: 2 (A: Protein, B: Protein)\nFinal Num. residues:  1239\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  419\nFinal Num. ligands or modified residues:  8\nFinal Num. water mol.:  411\nFinal Num. atoms:  6993\nModified residues found\nGLN A456\nGLN B457\nSmall mol ligands found\nPMP A457\nGOL A458\nGOL A459\nPMP B456\nGOL B458\nGOL B459\nStructure saved on /tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56/output_pdb_path.pdb\n\nINFO     /tmp/biobb/unitests_120/fix_chirality/fix_chirality_log.out:file_utils.py:409 Removed: [&amp;#x27;/tmp/biobb/unitests_120/fix_chirality/sandbox_d3d303f5-150e-45d5-92b3-bf0199adeb56&amp;#x27;]\nINFO     /tmp/biobb/unitests_120/fix_chirality/fix_chirality_log.out:file_utils.py:409\n\n--------------------------- Captured stdout teardown ---------------------------\n\nRemoving: /tmp/biobb/unitests_120\n&#34;}], &#34;biobb_model/test/unitests/test_model/test_fix_pdb.py::TestFixPdb::test_launch&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;biobb_model/test/unitests/test_model/test_fix_pdb.py::TestFixPdb::test_launch&#34;, &#34;duration&#34;: &#34;00:00:07&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;biobb_model/test/unitests/test_model/test_fix_pdb.py::TestFixPdb::test_launch&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:07&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;----------------------------- Captured stdout call -----------------------------\n2025-06-06 12:43:51,274 [MainThread  ] [INFO ]  Module: biobb_model.model.fix_pdb Version: 5.0.0\n126 VVIKVCEFQFCNDPFLGVYYHKNNKSWMESEFRVYSSANNCTFEYVSQPFLMDLEGKQGNFKNLREFVFKNIDGYFK-----IYSKH------TPINLVRDLPQGF-SALEPLVDLPIG----INIT------------------RFQTLLAL--------HRSYLT-PGDS------SSGWTAG------------------AAAYYVGYLQ------PRTFLLKYNENGTITDAVDCALDPLSETKCTLKSFTVEKGIYQTSNFRVQPTESIVRF-----PNITNLCPFGEVFNATRFASVYA------------WNRKR--------------------ISNCVADYSVL-YNSASFSTFKCYGV--------SPTKLNDLCFTNVYADSFVIRGDEVRQIAPGQTGKIADYNYKLPDDFTGCVIAWNSNNLDSKVGGNYNY---LYRLFRKSNLKPF\n    ||.|       |....|..|......|.    |.|........||.......|.|......|.|    ......|..     .|..|      .|..|..|....| ..|.||. .|.|    |..|                  .|.....|        ..|.|| |||.      ...|..|                  .|....|..|      ....||....|.....||.........|...||..    |.. .|.| ....|....|     .||....||.......|. .|..            |..||                    ......|||.. |.......|.....        .|....|.       ......|....|..  ..||.......| .||.|      |.|.|  ||....|   ||......|.|.|\n166 VVLK-------NEMARGYHYEDYGDYWR----RDYETEGSPDLEYSRDQLIKDVERIFAEIKPL----YEQLHAYVRTKLMDTYPFHISPTGCLPAHLLGDMWGRFWTNLYPLT-VPFGQKPNIDVTDAMLNQNWDAKRIFKEAEKFFVSIGLPNMTEGFWNNSMLTDPGDGRKVVCHPTAWDLGKGDFRIKMCTKVTMEDFLTAHHEMGHIQYDMAYASQPYLLRNGANEGFHEAVGEVMSLSVATPKHLKTM----GLL-SSDF-LEDNETEINFLFKQALNIVGTLPFTYMLEKWRW-MVFKGEIPKEEWMKKWWEMKRKIVGVVEPVPHDETYCDPASLFHVANDYSFIRYYTRTIFEFQFHEALCRIAKHDGPLHKCDI-------SNSTDAGEKLHQML--SVGKSQPWTSVL-KDFVG------SKNMD--VGPLLRYFEPLYTWLTEQNRKSF\n  Score=67\n\nNormalized score: 0.11185308848080133\nNot valid alignment\n19 STIEEQAKTFLDKFNHEAEDLFYQSSLASWNYNTNITEENVQNMNNAGDKWSAFLKEQSTLAQMYPLQEIQNLTVKLQLQALQQNGSSVLSEDKSKRLNTILNTMSTIYSTGKVCNPDNPQECLLLEPGLNEIMANSLDYNERLWAWESWRSEVGKQLRPLYEEYVVLKNEMARANHYEDYGDYWRGDYEVNGVDGYDYSRGQLIEDVEHTFEEIKPLYEHLHAYVRAKLMNAYPSYISPIGCLPAHLLGDMWGRFWTNLYSLTVPFGQKPNIDVTDAMVDQAWDAQRIFKEAEKFFVSVGLPNMTQGFWENSMLTDPGNVQKAVCHPTAWDLGKGDFRILMCTKVTMDDFLTAHHEMGHIQYDMAYAAQPFLLRNGANEGFHEAVGEIMSLSAATPKHLKSIGLLSPDFQEDNETEINFLLKQALTIVGTLPFTYMLEKWRWMVFKGEIPKDQWMKKWWEMKREIVGVVEPVPHDETYCDPASLFHVSNDYSFIRYYTRTLYQFQFQEALCQAAKHEGPLHKCDISNSTEAGQKLFNMLRLGKSEPWTLALENVVGAKNMNVRPLLNYFEPLFTWLKDQNKNSFVGWSTDWSPYAD\n   ||.|..||.|||.||.|||.|..|||||||.|||||..||||.|..||.|||.|...||.||....|.||.|.|||||||.|||.||.|||||||||||.|||.|||||||||||.|.||||||||||||..||..|.||||||||||.||.||||||||||||||||||||||..||||||||||.|||..|.....|||.|||.|||..|.|||||||.||||||.|||..||..|||.||||||||||||||||||||.|||||||||||||||||..|.|||.||||||||||||.||||||.|||.||||||||...|.||||||||||||||||.|||||||.|||||||||||||||||||.||.||||||||||||||||.||||.|||||||..||||.||.||||||||||.||||.|||||||||||||||||||||||||..||||||||||.|||||||||||||||||||||||.||||||||||||...|||.||||..|||.||||||||||||.||.||..||..|||.|||..|...||.|||.|.|||.|||||.|||..||..|||||.||||||||\n 1 STTEDLAKKFLDDFNSEAENLSHQSSLASWEYNTNISDENVQKMDEAGAKWSDFYEKQSKLAKNFSLEEIHNDTVKLQLQILQQSGSPVLSEDKSKRLNSILNAMSTIYSTGKVCKPNNPQECLLLEPGLDNIMGTSKDYNERLWAWEGWRAEVGKQLRPLYEEYVVLKNEMARGYHYEDYGDYWRRDYETEGSPDLEYSRDQLIKDVERIFAEIKPLYEQLHAYVRTKLMDTYPFHISPTGCLPAHLLGDMWGRFWTNLYPLTVPFGQKPNIDVTDAMLNQNWDAKRIFKEAEKFFVSIGLPNMTEGFWNNSMLTDPGDGRKVVCHPTAWDLGKGDFRIKMCTKVTMEDFLTAHHEMGHIQYDMAYASQPYLLRNGANEGFHEAVGEVMSLSVATPKHLKTMGLLSSDFLEDNETEINFLFKQALNIVGTLPFTYMLEKWRWMVFKGEIPKEEWMKKWWEMKRKIVGVVEPVPHDETYCDPASLFHVANDYSFIRYYTRTIFEFQFHEALCRIAKHDGPLHKCDISNSTDAGEKLHQMLSVGKSQPWTSVLKDFVGSKNMDVGPLLRYFEPLYTWLTEQNRKSFVGWNTDWSPYAD\n  Score=2643\n\nNormalized score: 4.412353923205342\n334 NLCPFGEVFNATRFASVYAWNRKRISNCVADYSVLYNSASFSTFKCYGVSPTKLNDLCFTNVYADSFVIRGDEVRQIAPGQTGKIADYNYKLPDDFTGCVIAWNSNNLDSKVGGNYNYLYRLFRKSNLKPFERDISTEIYQAGSTPCNGVEGFNCYFPLQSYGFQPTNGVGYQPYRVVVLSFELLHAPATVCGP\n    ||||||||||||.|.|||||.||.||||||||||||||..||||||||||.||||||||.||||||||..||.||||||||||.||||||||||||.|||.|||..|.|....|||||.||..|...|.|||||||.........||. ....|||.||..|||..|.|.|||||||||||||||.||||||||\n  1 NLCPFGEVFNATKFPSVYAWERKKISNCVADYSVLYNSTFFSTFKCYGVSATKLNDLCFSNVYADSFVVKGDDVRQIAPGQTGVIADYNYKLPDDFMGCVLAWNTRNIDATSTGNYNYKYRYLRHGKLRPFERDISNVPFSPDGKPCT-PPALNCYWPLNDYGFYTTTGIGYQPYRVVVLSFELLNAPATVCGP\n  Score=792\n\nNormalized score: 4.103626943005182\n84 PLQEIQNLTVKLQLQALQQNGSSVLSEDKSKRLN--------------TILNTM--STIYSTGKVCNPDNPQECLLLEPGLNEIMANSLDYNERL--------WAWESWRSEV---------------GKQLRPLYEEYVVLKN----------EMARANHYEDYGDYWRGDYEVNGVDGY\n   |..|..|.|......|......|....|.|...|              |.||..  |..|....|...|....   ..||.....|   |||..|        .||.......               || |||....   ..|          .....|.|....||  |.|...|. ||\n 4 PFGEVFNATKFPSVYAWERKKISNCVADYSVLYNSTFFSTFKCYGVSATKLNDLCFSNVYADSFVVKGDDVRQ---IAPGQTGVIA---DYNYKLPDDFMGCVLAWNTRNIDATSTGNYNYKYRYLRHGK-LRPFERD---ISNVPFSPDGKPCTPPALNCYWPLNDY--GFYTTTGI-GY\n  Score=46.5\n\nNormalized score: 0.24093264248704663\nNot valid alignment\nReference summary:\n   A -&amp;gt; Q9BYF1\n   B -&amp;gt; P0DTC2\nFixed :)\n2025-06-06 12:43:58,420 [MainThread  ] [INFO ]  Removed: []\n2025-06-06 12:43:58,422 [MainThread  ] [INFO ]  \nChecking if empty file: /tmp/biobb/unitests_120/fix_pdb/output_fixpdb.pdb\nComparing: \n        File_A: /tmp/biobb/unitests_120/fix_pdb/output_fixpdb.pdb\n        File_B: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_fixpdb.pdb\n        File_A hash: b&amp;#x27;g\\xebw_\\xb7\\xe0`\\x92\\xf8\\xc0\\xbd\\xc5\\xe6\\x86\\xbf\\xd6\\xc9\\x8f\\x17\\x82\\xfeV\\xd6\\xf7\\x89v\\xf6\\x11\\x80\\x96\\xfa_&amp;#x27;\n        File_B hash: b&amp;#x27;g\\xebw_\\xb7\\xe0`\\x92\\xf8\\xc0\\xbd\\xc5\\xe6\\x86\\xbf\\xd6\\xc9\\x8f\\x17\\x82\\xfeV\\xd6\\xf7\\x89v\\xf6\\x11\\x80\\x96\\xfa_&amp;#x27;\n\n------------------------------ Captured log call -------------------------------\nINFO     /tmp/biobb/unitests_120/fix_pdb/fix_pdb_log.out:file_utils.py:409 Module: biobb_model.model.fix_pdb Version: 5.0.0\nINFO     /tmp/biobb/unitests_120/fix_pdb/fix_pdb_log.out:file_utils.py:409 Removed: []\nINFO     /tmp/biobb/unitests_120/fix_pdb/fix_pdb_log.out:file_utils.py:409\n\n--------------------------- Captured stdout teardown ---------------------------\n\nRemoving: /tmp/biobb/unitests_120\n&#34;}], &#34;biobb_model/test/unitests/test_model/test_fix_side_chain.py::TestFixSideChain::test_launch&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;biobb_model/test/unitests/test_model/test_fix_side_chain.py::TestFixSideChain::test_launch&#34;, &#34;duration&#34;: &#34;00:00:04&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;biobb_model/test/unitests/test_model/test_fix_side_chain.py::TestFixSideChain::test_launch&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:04&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;----------------------------- Captured stdout call -----------------------------\n2025-06-06 12:43:58,477 [MainThread  ] [INFO ]  Module: biobb_model.model.fix_side_chain Version: 5.0.0\n2025-06-06 12:43:58,478 [MainThread  ] [INFO ]  /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40 directory successfully created\n2025-06-06 12:43:58,480 [MainThread  ] [INFO ]  Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40\n2025-06-06 12:43:58,481 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40/2ki5.pdb -o /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40/output_pdb_path.pdb --force_save fixside --fix ALL\n\n2025-06-06 12:44:01,492 [MainThread  ] [INFO ]  Executing: check_structure -i /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-...\n2025-06-06 12:44:01,492 [MainThread  ] [INFO ]  Exit code: 0\n2025-06-06 12:44:01,493 [MainThread  ] [INFO ]  ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40/2ki5.pdb loaded\n PDB id: 2KI5 \n Title: herpes simplex type-1 thymidine kinase in complex with the drug aciclovir at 1.9a resolution\n Experimental method: x-ray diffraction\n Keywords: transferase (phosphotransferase) thymid, transferase (phosphotransferase) thymidine kinase; viridae; ds-dna enveloped viruses; herpesviridae; alphaherpesvirinae antiviral drug (aciclovir)\n Resolution (A): 1.9\n\n Num. models: 1\n Num. chains: 2 (A: Protein, B: Protein)\n Num. residues:  908\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  296\n Num. ligands or modified residues:  4\n Num. water mol.:  292\n Num. atoms:  4961\nSmall mol ligands found\nSO4 A4\nAC2 A1\nSO4 B3\nAC2 B2\nRunning fixside. Options: --fix ALL\nFound 6 Residues with missing side chain atoms\n MET A46    CG,SD,CE\n PRO A154   CG,CD\n PRO A280   CG,CD\n MET B46    CG,SD,CE\n PRO B154   CG,CD\n ARG B220   CG,CD,NE,CZ,NH1,NH2\nFixing side chains\nMET A46\n  Adding new atom CG\n  Adding new atom SD\n  Adding new atom CE\nPRO A154\n  Adding new atom CG\n  Adding new atom CD\nPRO A280\n  Adding new atom CG\n  Adding new atom CD\nMET B46\n  Adding new atom CG\n  Adding new atom SD\n  Adding new atom CE\nPRO B154\n  Adding new atom CG\n  Adding new atom CD\nARG B220\n  Adding new atom CG\n  Adding new atom CD\n  Adding new atom NE\n  Adding new atom CZ\n  Adding new atom NH1\n  Adding new atom NH2\nFixed 6 side chain(s)\nChecking for steric clashes\n4 Severe Steric clashes/covalent bonds detected\n MET A46.CE   VAL A352.CG2    1.843 A\n MET A46.CE   THR A354.O      1.869 A\n MET B46.CE   VAL B352.O      0.921 A\n ARG B220.CZ  SO4 B3.O3       1.597 A\n5 Apolar steric clashes detected\n MET A46.CE   VAL A352.CG2    1.843 A\n MET A46.CE   THR A354.O      1.869 A\n MET B46.CE   VAL B352.O      0.921 A\n GLY B61.CA   ARG B220.NH1    2.301 A\n ARG B220.CZ  SO4 B3.O3       1.597 A\n3 Polar contacts (acceptors) detected\n MET A46.SD   VAL A352.O      2.171 A\n MET A46.SD   THR A354.O      2.569 A\n MET B46.SD   VAL B352.O      2.291 A\n2 Polar contacts (donors) detected\n GLY B61.N    ARG B220.NH1    2.443 A\n ARG B216.NH2 ARG B220.NH1    2.977 A\n1 Positively charged contacts detected\n ARG B216.NH2 ARG B220.NH1    2.977 A\nNo Negatively charged contacts detected\nFinal Num. models: 1\nFinal Num. chains: 2 (A: Protein, B: Protein)\nFinal Num. residues:  908\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  296\nFinal Num. ligands or modified residues:  4\nFinal Num. water mol.:  292\nFinal Num. atoms:  4979\nSmall mol ligands found\nSO4 A4\nAC2 A1\nSO4 B3\nAC2 B2\nStructure saved on /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40/output_pdb_path.pdb\n\n2025-06-06 12:44:01,495 [MainThread  ] [INFO ]  Removed: [&amp;#x27;/tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40&amp;#x27;]\n2025-06-06 12:44:01,496 [MainThread  ] [INFO ]  \nChecking if empty file: /tmp/biobb/unitests_120/fix_side_chain/output_pdb_path.pdb\nChecking RMSD between:\n     PDB_A: /tmp/biobb/unitests_120/fix_side_chain/output_pdb_path.pdb\n     PDB_B: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_pdb_path.pdb\n     Ignoring HETAMT in RMSD\n     Ignoring Hydrogen atoms in RMSD\n     Atoms ALIGNED in PDB_A: 4937\n     Atoms ALIGNED in PDB_B: 4937\n     RMS: 1.2641053867191999e-14\n     RMS_CUTOFF: 1\n\n------------------------------ Captured log call -------------------------------\nINFO     /tmp/biobb/unitests_120/fix_side_chain/fix_side_chain_log.out:file_utils.py:409 Module: biobb_model.model.fix_side_chain Version: 5.0.0\nINFO     /tmp/biobb/unitests_120/fix_side_chain/fix_side_chain_log.out:file_utils.py:82 /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40 directory successfully created\nINFO     /tmp/biobb/unitests_120/fix_side_chain/fix_side_chain_log.out:file_utils.py:409 Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40\nINFO     /tmp/biobb/unitests_120/fix_side_chain/fix_side_chain_log.out:cmd_wrapper.py:63 check_structure -i /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40/2ki5.pdb -o /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40/output_pdb_path.pdb --force_save fixside --fix ALL\n\nINFO     /tmp/biobb/unitests_120/fix_side_chain/fix_side_chain_log.out:cmd_wrapper.py:36 Executing: check_structure -i /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-...\nINFO     /tmp/biobb/unitests_120/fix_side_chain/fix_side_chain_log.out:cmd_wrapper.py:37 Exit code: 0\nINFO     /tmp/biobb/unitests_120/fix_side_chain/fix_side_chain_log.out:cmd_wrapper.py:41 ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40/2ki5.pdb loaded\n PDB id: 2KI5 \n Title: herpes simplex type-1 thymidine kinase in complex with the drug aciclovir at 1.9a resolution\n Experimental method: x-ray diffraction\n Keywords: transferase (phosphotransferase) thymid, transferase (phosphotransferase) thymidine kinase; viridae; ds-dna enveloped viruses; herpesviridae; alphaherpesvirinae antiviral drug (aciclovir)\n Resolution (A): 1.9\n\n Num. models: 1\n Num. chains: 2 (A: Protein, B: Protein)\n Num. residues:  908\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  296\n Num. ligands or modified residues:  4\n Num. water mol.:  292\n Num. atoms:  4961\nSmall mol ligands found\nSO4 A4\nAC2 A1\nSO4 B3\nAC2 B2\nRunning fixside. Options: --fix ALL\nFound 6 Residues with missing side chain atoms\n MET A46    CG,SD,CE\n PRO A154   CG,CD\n PRO A280   CG,CD\n MET B46    CG,SD,CE\n PRO B154   CG,CD\n ARG B220   CG,CD,NE,CZ,NH1,NH2\nFixing side chains\nMET A46\n  Adding new atom CG\n  Adding new atom SD\n  Adding new atom CE\nPRO A154\n  Adding new atom CG\n  Adding new atom CD\nPRO A280\n  Adding new atom CG\n  Adding new atom CD\nMET B46\n  Adding new atom CG\n  Adding new atom SD\n  Adding new atom CE\nPRO B154\n  Adding new atom CG\n  Adding new atom CD\nARG B220\n  Adding new atom CG\n  Adding new atom CD\n  Adding new atom NE\n  Adding new atom CZ\n  Adding new atom NH1\n  Adding new atom NH2\nFixed 6 side chain(s)\nChecking for steric clashes\n4 Severe Steric clashes/covalent bonds detected\n MET A46.CE   VAL A352.CG2    1.843 A\n MET A46.CE   THR A354.O      1.869 A\n MET B46.CE   VAL B352.O      0.921 A\n ARG B220.CZ  SO4 B3.O3       1.597 A\n5 Apolar steric clashes detected\n MET A46.CE   VAL A352.CG2    1.843 A\n MET A46.CE   THR A354.O      1.869 A\n MET B46.CE   VAL B352.O      0.921 A\n GLY B61.CA   ARG B220.NH1    2.301 A\n ARG B220.CZ  SO4 B3.O3       1.597 A\n3 Polar contacts (acceptors) detected\n MET A46.SD   VAL A352.O      2.171 A\n MET A46.SD   THR A354.O      2.569 A\n MET B46.SD   VAL B352.O      2.291 A\n2 Polar contacts (donors) detected\n GLY B61.N    ARG B220.NH1    2.443 A\n ARG B216.NH2 ARG B220.NH1    2.977 A\n1 Positively charged contacts detected\n ARG B216.NH2 ARG B220.NH1    2.977 A\nNo Negatively charged contacts detected\nFinal Num. models: 1\nFinal Num. chains: 2 (A: Protein, B: Protein)\nFinal Num. residues:  908\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  296\nFinal Num. ligands or modified residues:  4\nFinal Num. water mol.:  292\nFinal Num. atoms:  4979\nSmall mol ligands found\nSO4 A4\nAC2 A1\nSO4 B3\nAC2 B2\nStructure saved on /tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40/output_pdb_path.pdb\n\nINFO     /tmp/biobb/unitests_120/fix_side_chain/fix_side_chain_log.out:file_utils.py:409 Removed: [&amp;#x27;/tmp/biobb/unitests_120/fix_side_chain/sandbox_17123c96-a128-44b3-b8a8-c440fd4c8f40&amp;#x27;]\nINFO     /tmp/biobb/unitests_120/fix_side_chain/fix_side_chain_log.out:file_utils.py:409\n\n--------------------------- Captured stdout teardown ---------------------------\n\nRemoving: /tmp/biobb/unitests_120\n&#34;}], &#34;biobb_model/test/unitests/test_model/test_fix_ssbonds.py::TestFixSSBonds::test_launch&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;biobb_model/test/unitests/test_model/test_fix_ssbonds.py::TestFixSSBonds::test_launch&#34;, &#34;duration&#34;: &#34;00:00:02&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;biobb_model/test/unitests/test_model/test_fix_ssbonds.py::TestFixSSBonds::test_launch&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:02&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;----------------------------- Captured stdout call -----------------------------\n2025-06-06 12:44:02,074 [MainThread  ] [INFO ]  Module: biobb_model.model.fix_ssbonds Version: 5.0.0\n2025-06-06 12:44:02,075 [MainThread  ] [INFO ]  /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193 directory successfully created\n2025-06-06 12:44:02,076 [MainThread  ] [INFO ]  Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/1aki.pdb to /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193\n2025-06-06 12:44:02,077 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193/1aki.pdb -o /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193/output_pdb_path.pdb --force_save --non_interactive getss --mark All\n\n2025-06-06 12:44:03,715 [MainThread  ] [INFO ]  Executing: check_structure -i /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-465...\n2025-06-06 12:44:03,716 [MainThread  ] [INFO ]  Exit code: 0\n2025-06-06 12:44:03,716 [MainThread  ] [INFO ]  ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193/1aki.pdb loaded\n PDB id: 1AKI \n Title: the structure of the orthorhombic form of hen egg-white lysozyme at 1.5 angstroms resolution\n Experimental method: x-ray diffraction\n Keywords: hydrolase, glycosidase\n Resolution (A): 1.5\n\n Num. models: 1\n Num. chains: 1 (A: Protein)\n Num. residues:  207\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  78\n Num. ligands or modified residues:  0\n Num. water mol.:  78\n Num. atoms:  1079\nRunning getss. Options: --mark All\nDetected 4 Possible SS Bonds\n CYS A6.SG    CYS A127.SG     3.880\n CYS A30.SG   CYS A115.SG     4.005\n CYS A64.SG   CYS A80.SG      3.948\n CYS A76.SG   CYS A94.SG      4.073\nFinal Num. models: 1\nFinal Num. chains: 1 (A: Protein)\nFinal Num. residues:  207\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  78\nFinal Num. ligands or modified residues:  0\nFinal Num. water mol.:  78\nFinal Num. atoms:  1079\nStructure saved on /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193/output_pdb_path.pdb\n\n2025-06-06 12:44:03,718 [MainThread  ] [INFO ]  Removed: [&amp;#x27;/tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193&amp;#x27;]\n2025-06-06 12:44:03,719 [MainThread  ] [INFO ]  \nChecking if empty file: /tmp/biobb/unitests_120/fix_ssbonds/output_pdb_path.pdb\nChecking RMSD between:\n     PDB_A: /tmp/biobb/unitests_120/fix_ssbonds/output_pdb_path.pdb\n     PDB_B: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_ssbonds.pdb\n     Ignoring HETAMT in RMSD\n     Ignoring Hydrogen atoms in RMSD\n     Atoms ALIGNED in PDB_A: 1079\n     Atoms ALIGNED in PDB_B: 1079\n     RMS: 4.6416389346120765e-15\n     RMS_CUTOFF: 1\n\n------------------------------ Captured log call -------------------------------\nINFO     /tmp/biobb/unitests_120/fix_ssbonds/fix_ssbonds_log.out:file_utils.py:409 Module: biobb_model.model.fix_ssbonds Version: 5.0.0\nINFO     /tmp/biobb/unitests_120/fix_ssbonds/fix_ssbonds_log.out:file_utils.py:82 /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193 directory successfully created\nINFO     /tmp/biobb/unitests_120/fix_ssbonds/fix_ssbonds_log.out:file_utils.py:409 Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/1aki.pdb to /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193\nINFO     /tmp/biobb/unitests_120/fix_ssbonds/fix_ssbonds_log.out:cmd_wrapper.py:63 check_structure -i /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193/1aki.pdb -o /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193/output_pdb_path.pdb --force_save --non_interactive getss --mark All\n\nINFO     /tmp/biobb/unitests_120/fix_ssbonds/fix_ssbonds_log.out:cmd_wrapper.py:36 Executing: check_structure -i /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-465...\nINFO     /tmp/biobb/unitests_120/fix_ssbonds/fix_ssbonds_log.out:cmd_wrapper.py:37 Exit code: 0\nINFO     /tmp/biobb/unitests_120/fix_ssbonds/fix_ssbonds_log.out:cmd_wrapper.py:41 ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193/1aki.pdb loaded\n PDB id: 1AKI \n Title: the structure of the orthorhombic form of hen egg-white lysozyme at 1.5 angstroms resolution\n Experimental method: x-ray diffraction\n Keywords: hydrolase, glycosidase\n Resolution (A): 1.5\n\n Num. models: 1\n Num. chains: 1 (A: Protein)\n Num. residues:  207\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  78\n Num. ligands or modified residues:  0\n Num. water mol.:  78\n Num. atoms:  1079\nRunning getss. Options: --mark All\nDetected 4 Possible SS Bonds\n CYS A6.SG    CYS A127.SG     3.880\n CYS A30.SG   CYS A115.SG     4.005\n CYS A64.SG   CYS A80.SG      3.948\n CYS A76.SG   CYS A94.SG      4.073\nFinal Num. models: 1\nFinal Num. chains: 1 (A: Protein)\nFinal Num. residues:  207\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  78\nFinal Num. ligands or modified residues:  0\nFinal Num. water mol.:  78\nFinal Num. atoms:  1079\nStructure saved on /tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193/output_pdb_path.pdb\n\nINFO     /tmp/biobb/unitests_120/fix_ssbonds/fix_ssbonds_log.out:file_utils.py:409 Removed: [&amp;#x27;/tmp/biobb/unitests_120/fix_ssbonds/sandbox_d2761ea3-996d-4657-a679-c390a2a17193&amp;#x27;]\nINFO     /tmp/biobb/unitests_120/fix_ssbonds/fix_ssbonds_log.out:file_utils.py:409\n\n--------------------------- Captured stdout teardown ---------------------------\n\nRemoving: /tmp/biobb/unitests_120\n&#34;}], &#34;biobb_model/test/unitests/test_model/test_mutate.py::TestFixSideChain::test_launch&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Passed&#34;, &#34;testId&#34;: &#34;biobb_model/test/unitests/test_model/test_mutate.py::TestFixSideChain::test_launch&#34;, &#34;duration&#34;: &#34;00:00:03&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Passed&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;biobb_model/test/unitests/test_model/test_mutate.py::TestFixSideChain::test_launch&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:00:03&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;----------------------------- Captured stdout call -----------------------------\n2025-06-06 12:44:03,867 [MainThread  ] [INFO ]  Module: biobb_model.model.mutate Version: 5.0.0\n2025-06-06 12:44:03,867 [MainThread  ] [INFO ]  /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8 directory successfully created\n2025-06-06 12:44:03,869 [MainThread  ] [INFO ]  Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8\n2025-06-06 12:44:03,869 [MainThread  ] [INFO ]  check_structure -i /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8/2ki5.pdb -o /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8/output_mutated_pdb_path.pdb --force_save --non_interactive mutateside --mut Leu49Ile,B:arg51Lys\n\n2025-06-06 12:44:06,630 [MainThread  ] [INFO ]  Executing: check_structure -i /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b21...\n2025-06-06 12:44:06,630 [MainThread  ] [INFO ]  Exit code: 0\n2025-06-06 12:44:06,631 [MainThread  ] [INFO ]  ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8/2ki5.pdb loaded\n PDB id: 2KI5 \n Title: herpes simplex type-1 thymidine kinase in complex with the drug aciclovir at 1.9a resolution\n Experimental method: x-ray diffraction\n Keywords: transferase (phosphotransferase) thymid, transferase (phosphotransferase) thymidine kinase; viridae; ds-dna enveloped viruses; herpesviridae; alphaherpesvirinae antiviral drug (aciclovir)\n Resolution (A): 1.9\n\n Num. models: 1\n Num. chains: 2 (A: Protein, B: Protein)\n Num. residues:  908\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  296\n Num. ligands or modified residues:  4\n Num. water mol.:  292\n Num. atoms:  4961\nSmall mol ligands found\nSO4 A4\nAC2 A1\nSO4 B3\nAC2 B2\nRunning mutateside. Options: --mut Leu49Ile,B:arg51Lys\nMutations to perform\n*:LEU49ILE\nB:ARG51LYS\nReplacing LEU A49 into ILE\n  Deleting CD2\n  Renaming CG to CG1\n  Adding new atom CG2\nReplacing LEU B49 into ILE\n  Deleting CD2\n  Renaming CG to CG1\n  Adding new atom CG2\n\nReplacing ARG B51 into LYS\n  Deleting NE\n  Deleting CZ\n  Deleting NH1\n  Deleting NH2\n  Adding new atom CE\n  Adding new atom NZ\n\nChecking for steric clashes\nNo Severe Steric clashes/covalent bonds detected\n4 Apolar steric clashes detected\n ILE A49.CG2  THR A350.OG1    2.815 A\n ILE B49.CG2  THR B350.OG1    2.611 A\n LYS B51.NZ   LEU B139.CG     2.403 A\n LYS B51.NZ   ILE B143.CD1    2.319 A\nNo Polar contacts (acceptors) detected\nNo Polar contacts (donors) detected\nNo Positively charged contacts detected\nNo Negatively charged contacts detected\nFinal Num. models: 1\nFinal Num. chains: 2 (A: Protein, B: Protein)\nFinal Num. residues:  908\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  296\nFinal Num. ligands or modified residues:  4\nFinal Num. water mol.:  292\nFinal Num. atoms:  4959\nSmall mol ligands found\nSO4 A4\nAC2 A1\nSO4 B3\nAC2 B2\nStructure saved on /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8/output_mutated_pdb_path.pdb\n\n2025-06-06 12:44:06,633 [MainThread  ] [INFO ]  Removed: [&amp;#x27;/tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8&amp;#x27;]\n2025-06-06 12:44:06,634 [MainThread  ] [INFO ]  \nChecking if empty file: /tmp/biobb/unitests_120/mutate/output_mutated_pdb_path.pdb\nChecking RMSD between:\n     PDB_A: /tmp/biobb/unitests_120/mutate/output_mutated_pdb_path.pdb\n     PDB_B: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/reference/model/output_mutated_pdb_path.pdb\n     Ignoring HETAMT in RMSD\n     Ignoring Hydrogen atoms in RMSD\n     Atoms ALIGNED in PDB_A: 4917\n     Atoms ALIGNED in PDB_B: 4917\n     RMS: 1.2079768858320278e-14\n     RMS_CUTOFF: 1\n\n------------------------------ Captured log call -------------------------------\nINFO     /tmp/biobb/unitests_120/mutate/mutate_log.out:file_utils.py:409 Module: biobb_model.model.mutate Version: 5.0.0\nINFO     /tmp/biobb/unitests_120/mutate/mutate_log.out:file_utils.py:82 /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8 directory successfully created\nINFO     /tmp/biobb/unitests_120/mutate/mutate_log.out:file_utils.py:409 Copy: /home/gbayarri/bxcl/_work/biobb_model/biobb_model/biobb_model/test/data/model/2ki5.pdb to /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8\nINFO     /tmp/biobb/unitests_120/mutate/mutate_log.out:cmd_wrapper.py:63 check_structure -i /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8/2ki5.pdb -o /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8/output_mutated_pdb_path.pdb --force_save --non_interactive mutateside --mut Leu49Ile,B:arg51Lys\n\nINFO     /tmp/biobb/unitests_120/mutate/mutate_log.out:cmd_wrapper.py:36 Executing: check_structure -i /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b21...\nINFO     /tmp/biobb/unitests_120/mutate/mutate_log.out:cmd_wrapper.py:37 Exit code: 0\nINFO     /tmp/biobb/unitests_120/mutate/mutate_log.out:cmd_wrapper.py:41 ================================================================================\n=                   BioBB structure checking utility v3.15.6                   =\n=            P. Andrio, A. Hospital, G. Bayarri, J.L. Gelpi 2018-23            =\n================================================================================\n\nWarning: sequence features may not be available, use --sequence for  external fasta input\nStructure /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8/2ki5.pdb loaded\n PDB id: 2KI5 \n Title: herpes simplex type-1 thymidine kinase in complex with the drug aciclovir at 1.9a resolution\n Experimental method: x-ray diffraction\n Keywords: transferase (phosphotransferase) thymid, transferase (phosphotransferase) thymidine kinase; viridae; ds-dna enveloped viruses; herpesviridae; alphaherpesvirinae antiviral drug (aciclovir)\n Resolution (A): 1.9\n\n Num. models: 1\n Num. chains: 2 (A: Protein, B: Protein)\n Num. residues:  908\n Num. residues with ins. codes:  0\n Num. residues with H atoms: 0\n Num. HETATM residues:  296\n Num. ligands or modified residues:  4\n Num. water mol.:  292\n Num. atoms:  4961\nSmall mol ligands found\nSO4 A4\nAC2 A1\nSO4 B3\nAC2 B2\nRunning mutateside. Options: --mut Leu49Ile,B:arg51Lys\nMutations to perform\n*:LEU49ILE\nB:ARG51LYS\nReplacing LEU A49 into ILE\n  Deleting CD2\n  Renaming CG to CG1\n  Adding new atom CG2\nReplacing LEU B49 into ILE\n  Deleting CD2\n  Renaming CG to CG1\n  Adding new atom CG2\n\nReplacing ARG B51 into LYS\n  Deleting NE\n  Deleting CZ\n  Deleting NH1\n  Deleting NH2\n  Adding new atom CE\n  Adding new atom NZ\n\nChecking for steric clashes\nNo Severe Steric clashes/covalent bonds detected\n4 Apolar steric clashes detected\n ILE A49.CG2  THR A350.OG1    2.815 A\n ILE B49.CG2  THR B350.OG1    2.611 A\n LYS B51.NZ   LEU B139.CG     2.403 A\n LYS B51.NZ   ILE B143.CD1    2.319 A\nNo Polar contacts (acceptors) detected\nNo Polar contacts (donors) detected\nNo Positively charged contacts detected\nNo Negatively charged contacts detected\nFinal Num. models: 1\nFinal Num. chains: 2 (A: Protein, B: Protein)\nFinal Num. residues:  908\nFinal Num. residues with ins. codes:  0\nFinal Num. residues with H atoms: 0\nFinal Num. HETATM residues:  296\nFinal Num. ligands or modified residues:  4\nFinal Num. water mol.:  292\nFinal Num. atoms:  4959\nSmall mol ligands found\nSO4 A4\nAC2 A1\nSO4 B3\nAC2 B2\nStructure saved on /tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8/output_mutated_pdb_path.pdb\n\nINFO     /tmp/biobb/unitests_120/mutate/mutate_log.out:file_utils.py:409 Removed: [&amp;#x27;/tmp/biobb/unitests_120/mutate/sandbox_f95f7467-72dc-4834-b218-293c55844cd8&amp;#x27;]\nINFO     /tmp/biobb/unitests_120/mutate/mutate_log.out:file_utils.py:409\n\n--------------------------- Captured stdout teardown ---------------------------\n\nRemoving: /tmp/biobb/unitests_120\n&#34;}]}, &#34;renderCollapsed&#34;: [&#34;passed&#34;], &#34;initialSort&#34;: &#34;result&#34;, &#34;title&#34;: &#34;report.html&#34;}"></div>
    <script>
      (function(){function r(e,n,t){function o(i,f){if(!n[i]){if(!e[i]){var c="function"==typeof require&&require;if(!f&&c)return c(i,!0);if(u)return u(i,!0);var a=new Error("Cannot find module '"+i+"'");throw a.code="MODULE_NOT_FOUND",a}var p=n[i]={exports:{}};e[i][0].call(p.exports,function(r){var n=e[i][1][r];return o(n||r)},p,p.exports,r,e,n,t)}return n[i].exports}for(var u="function"==typeof require&&require,i=0;i<t.length;i++)o(t[i]);return o}return r})()({1:[function(require,module,exports){
const { getCollapsedCategory, setCollapsedIds } = require('./storage.js')

class DataManager {
    setManager(data) {
        const collapsedCategories = [...getCollapsedCategory(data.renderCollapsed)]
        const collapsedIds = []
        const tests = Object.values(data.tests).flat().map((test, index) => {
            const collapsed = collapsedCategories.includes(test.result.toLowerCase())
            const id = `test_${index}`
            if (collapsed) {
                collapsedIds.push(id)
            }
            return {
                ...test,
                id,
                collapsed,
            }
        })
        const dataBlob = { ...data, tests }
        this.data = { ...dataBlob }
        this.renderData = { ...dataBlob }
        setCollapsedIds(collapsedIds)
    }

    get allData() {
        return { ...this.data }
    }

    resetRender() {
        this.renderData = { ...this.data }
    }

    setRender(data) {
        this.renderData.tests = [...data]
    }

    toggleCollapsedItem(id) {
        this.renderData.tests = this.renderData.tests.map((test) =>
            test.id === id ? { ...test, collapsed: !test.collapsed } : test,
        )
    }

    set allCollapsed(collapsed) {
        this.renderData = { ...this.renderData, tests: [...this.renderData.tests.map((test) => (
            { ...test, collapsed }
        ))] }
    }

    get testSubset() {
        return [...this.renderData.tests]
    }

    get environment() {
        return this.renderData.environment
    }

    get initialSort() {
        return this.data.initialSort
    }
}

module.exports = {
    manager: new DataManager(),
}

},{"./storage.js":8}],2:[function(require,module,exports){
const mediaViewer = require('./mediaviewer.js')
const templateEnvRow = document.getElementById('template_environment_row')
const templateResult = document.getElementById('template_results-table__tbody')

function htmlToElements(html) {
    const temp = document.createElement('template')
    temp.innerHTML = html
    return temp.content.childNodes
}

const find = (selector, elem) => {
    if (!elem) {
        elem = document
    }
    return elem.querySelector(selector)
}

const findAll = (selector, elem) => {
    if (!elem) {
        elem = document
    }
    return [...elem.querySelectorAll(selector)]
}

const dom = {
    getStaticRow: (key, value) => {
        const envRow = templateEnvRow.content.cloneNode(true)
        const isObj = typeof value === 'object' && value !== null
        const values = isObj ? Object.keys(value).map((k) => `${k}: ${value[k]}`) : null

        const valuesElement = htmlToElements(
            values ? `<ul>${values.map((val) => `<li>${val}</li>`).join('')}<ul>` : `<div>${value}</div>`)[0]
        const td = findAll('td', envRow)
        td[0].textContent = key
        td[1].appendChild(valuesElement)

        return envRow
    },
    getResultTBody: ({ testId, id, log, extras, resultsTableRow, tableHtml, result, collapsed }) => {
        const resultBody = templateResult.content.cloneNode(true)
        resultBody.querySelector('tbody').classList.add(result.toLowerCase())
        resultBody.querySelector('tbody').id = testId
        resultBody.querySelector('.collapsible').dataset.id = id

        resultsTableRow.forEach((html) => {
            const t = document.createElement('template')
            t.innerHTML = html
            resultBody.querySelector('.collapsible').appendChild(t.content)
        })

        if (log) {
            // Wrap lines starting with "E" with span.error to color those lines red
            const wrappedLog = log.replace(/^E.*$/gm, (match) => `<span class="error">${match}</span>`)
            resultBody.querySelector('.log').innerHTML = wrappedLog
        } else {
            resultBody.querySelector('.log').remove()
        }

        if (collapsed) {
            resultBody.querySelector('.collapsible > td')?.classList.add('collapsed')
            resultBody.querySelector('.extras-row').classList.add('hidden')
        } else {
            resultBody.querySelector('.collapsible > td')?.classList.remove('collapsed')
        }

        const media = []
        extras?.forEach(({ name, format_type, content }) => {
            if (['image', 'video'].includes(format_type)) {
                media.push({ path: content, name, format_type })
            }

            if (format_type === 'html') {
                resultBody.querySelector('.extraHTML').insertAdjacentHTML('beforeend', `<div>${content}</div>`)
            }
        })
        mediaViewer.setup(resultBody, media)

        // Add custom html from the pytest_html_results_table_html hook
        tableHtml?.forEach((item) => {
            resultBody.querySelector('td[class="extra"]').insertAdjacentHTML('beforeend', item)
        })

        return resultBody
    },
}

module.exports = {
    dom,
    htmlToElements,
    find,
    findAll,
}

},{"./mediaviewer.js":6}],3:[function(require,module,exports){
const { manager } = require('./datamanager.js')
const { doSort } = require('./sort.js')
const storageModule = require('./storage.js')

const getFilteredSubSet = (filter) =>
    manager.allData.tests.filter(({ result }) => filter.includes(result.toLowerCase()))

const doInitFilter = () => {
    const currentFilter = storageModule.getVisible()
    const filteredSubset = getFilteredSubSet(currentFilter)
    manager.setRender(filteredSubset)
}

const doFilter = (type, show) => {
    if (show) {
        storageModule.showCategory(type)
    } else {
        storageModule.hideCategory(type)
    }

    const currentFilter = storageModule.getVisible()
    const filteredSubset = getFilteredSubSet(currentFilter)
    manager.setRender(filteredSubset)

    const sortColumn = storageModule.getSort()
    doSort(sortColumn, true)
}

module.exports = {
    doFilter,
    doInitFilter,
}

},{"./datamanager.js":1,"./sort.js":7,"./storage.js":8}],4:[function(require,module,exports){
const { redraw, bindEvents, renderStatic } = require('./main.js')
const { doInitFilter } = require('./filter.js')
const { doInitSort } = require('./sort.js')
const { manager } = require('./datamanager.js')
const data = JSON.parse(document.getElementById('data-container').dataset.jsonblob)

function init() {
    manager.setManager(data)
    doInitFilter()
    doInitSort()
    renderStatic()
    redraw()
    bindEvents()
}

init()

},{"./datamanager.js":1,"./filter.js":3,"./main.js":5,"./sort.js":7}],5:[function(require,module,exports){
const { dom, find, findAll } = require('./dom.js')
const { manager } = require('./datamanager.js')
const { doSort } = require('./sort.js')
const { doFilter } = require('./filter.js')
const {
    getVisible,
    getCollapsedIds,
    setCollapsedIds,
    getSort,
    getSortDirection,
    possibleFilters,
} = require('./storage.js')

const removeChildren = (node) => {
    while (node.firstChild) {
        node.removeChild(node.firstChild)
    }
}

const renderStatic = () => {
    const renderEnvironmentTable = () => {
        const environment = manager.environment
        const rows = Object.keys(environment).map((key) => dom.getStaticRow(key, environment[key]))
        const table = document.getElementById('environment')
        removeChildren(table)
        rows.forEach((row) => table.appendChild(row))
    }
    renderEnvironmentTable()
}

const addItemToggleListener = (elem) => {
    elem.addEventListener('click', ({ target }) => {
        const id = target.parentElement.dataset.id
        manager.toggleCollapsedItem(id)

        const collapsedIds = getCollapsedIds()
        if (collapsedIds.includes(id)) {
            const updated = collapsedIds.filter((item) => item !== id)
            setCollapsedIds(updated)
        } else {
            collapsedIds.push(id)
            setCollapsedIds(collapsedIds)
        }
        redraw()
    })
}

const renderContent = (tests) => {
    const sortAttr = getSort(manager.initialSort)
    const sortAsc = JSON.parse(getSortDirection())
    const rows = tests.map(dom.getResultTBody)
    const table = document.getElementById('results-table')
    const tableHeader = document.getElementById('results-table-head')

    const newTable = document.createElement('table')
    newTable.id = 'results-table'

    // remove all sorting classes and set the relevant
    findAll('.sortable', tableHeader).forEach((elem) => elem.classList.remove('asc', 'desc'))
    tableHeader.querySelector(`.sortable[data-column-type="${sortAttr}"]`)?.classList.add(sortAsc ? 'desc' : 'asc')
    newTable.appendChild(tableHeader)

    if (!rows.length) {
        const emptyTable = document.getElementById('template_results-table__body--empty').content.cloneNode(true)
        newTable.appendChild(emptyTable)
    } else {
        rows.forEach((row) => {
            if (!!row) {
                findAll('.collapsible td:not(.col-links', row).forEach(addItemToggleListener)
                find('.logexpander', row).addEventListener('click',
                    (evt) => evt.target.parentNode.classList.toggle('expanded'),
                )
                newTable.appendChild(row)
            }
        })
    }

    table.replaceWith(newTable)
}

const renderDerived = () => {
    const currentFilter = getVisible()
    possibleFilters.forEach((result) => {
        const input = document.querySelector(`input[data-test-result="${result}"]`)
        input.checked = currentFilter.includes(result)
    })
}

const bindEvents = () => {
    const filterColumn = (evt) => {
        const { target: element } = evt
        const { testResult } = element.dataset

        doFilter(testResult, element.checked)
        const collapsedIds = getCollapsedIds()
        const updated = manager.renderData.tests.map((test) => {
            return {
                ...test,
                collapsed: collapsedIds.includes(test.id),
            }
        })
        manager.setRender(updated)
        redraw()
    }

    const header = document.getElementById('environment-header')
    header.addEventListener('click', () => {
        const table = document.getElementById('environment')
        table.classList.toggle('hidden')
        header.classList.toggle('collapsed')
    })

    findAll('input[name="filter_checkbox"]').forEach((elem) => {
        elem.addEventListener('click', filterColumn)
    })

    findAll('.sortable').forEach((elem) => {
        elem.addEventListener('click', (evt) => {
            const { target: element } = evt
            const { columnType } = element.dataset
            doSort(columnType)
            redraw()
        })
    })

    document.getElementById('show_all_details').addEventListener('click', () => {
        manager.allCollapsed = false
        setCollapsedIds([])
        redraw()
    })
    document.getElementById('hide_all_details').addEventListener('click', () => {
        manager.allCollapsed = true
        const allIds = manager.renderData.tests.map((test) => test.id)
        setCollapsedIds(allIds)
        redraw()
    })
}

const redraw = () => {
    const { testSubset } = manager

    renderContent(testSubset)
    renderDerived()
}

module.exports = {
    redraw,
    bindEvents,
    renderStatic,
}

},{"./datamanager.js":1,"./dom.js":2,"./filter.js":3,"./sort.js":7,"./storage.js":8}],6:[function(require,module,exports){
class MediaViewer {
    constructor(assets) {
        this.assets = assets
        this.index = 0
    }

    nextActive() {
        this.index = this.index === this.assets.length - 1 ? 0 : this.index + 1
        return [this.activeFile, this.index]
    }

    prevActive() {
        this.index = this.index === 0 ? this.assets.length - 1 : this.index -1
        return [this.activeFile, this.index]
    }

    get currentIndex() {
        return this.index
    }

    get activeFile() {
        return this.assets[this.index]
    }
}


const setup = (resultBody, assets) => {
    if (!assets.length) {
        resultBody.querySelector('.media').classList.add('hidden')
        return
    }

    const mediaViewer = new MediaViewer(assets)
    const container = resultBody.querySelector('.media-container')
    const leftArrow = resultBody.querySelector('.media-container__nav--left')
    const rightArrow = resultBody.querySelector('.media-container__nav--right')
    const mediaName = resultBody.querySelector('.media__name')
    const counter = resultBody.querySelector('.media__counter')
    const imageEl = resultBody.querySelector('img')
    const sourceEl = resultBody.querySelector('source')
    const videoEl = resultBody.querySelector('video')

    const setImg = (media, index) => {
        if (media?.format_type === 'image') {
            imageEl.src = media.path

            imageEl.classList.remove('hidden')
            videoEl.classList.add('hidden')
        } else if (media?.format_type === 'video') {
            sourceEl.src = media.path

            videoEl.classList.remove('hidden')
            imageEl.classList.add('hidden')
        }

        mediaName.innerText = media?.name
        counter.innerText = `${index + 1} / ${assets.length}`
    }
    setImg(mediaViewer.activeFile, mediaViewer.currentIndex)

    const moveLeft = () => {
        const [media, index] = mediaViewer.prevActive()
        setImg(media, index)
    }
    const doRight = () => {
        const [media, index] = mediaViewer.nextActive()
        setImg(media, index)
    }
    const openImg = () => {
        window.open(mediaViewer.activeFile.path, '_blank')
    }
    if (assets.length === 1) {
        container.classList.add('media-container--fullscreen')
    } else {
        leftArrow.addEventListener('click', moveLeft)
        rightArrow.addEventListener('click', doRight)
    }
    imageEl.addEventListener('click', openImg)
}

module.exports = {
    setup,
}

},{}],7:[function(require,module,exports){
const { manager } = require('./datamanager.js')
const storageModule = require('./storage.js')

const genericSort = (list, key, ascending, customOrder) => {
    let sorted
    if (customOrder) {
        sorted = list.sort((a, b) => {
            const aValue = a.result.toLowerCase()
            const bValue = b.result.toLowerCase()

            const aIndex = customOrder.findIndex((item) => item.toLowerCase() === aValue)
            const bIndex = customOrder.findIndex((item) => item.toLowerCase() === bValue)

            // Compare the indices to determine the sort order
            return aIndex - bIndex
        })
    } else {
        sorted = list.sort((a, b) => a[key] === b[key] ? 0 : a[key] > b[key] ? 1 : -1)
    }

    if (ascending) {
        sorted.reverse()
    }
    return sorted
}

const durationSort = (list, ascending) => {
    const parseDuration = (duration) => {
        if (duration.includes(':')) {
            // If it's in the format "HH:mm:ss"
            const [hours, minutes, seconds] = duration.split(':').map(Number)
            return (hours * 3600 + minutes * 60 + seconds) * 1000
        } else {
            // If it's in the format "nnn ms"
            return parseInt(duration)
        }
    }
    const sorted = list.sort((a, b) => parseDuration(a['duration']) - parseDuration(b['duration']))
    if (ascending) {
        sorted.reverse()
    }
    return sorted
}

const doInitSort = () => {
    const type = storageModule.getSort(manager.initialSort)
    const ascending = storageModule.getSortDirection()
    const list = manager.testSubset
    const initialOrder = ['Error', 'Failed', 'Rerun', 'XFailed', 'XPassed', 'Skipped', 'Passed']

    storageModule.setSort(type)
    storageModule.setSortDirection(ascending)

    if (type?.toLowerCase() === 'original') {
        manager.setRender(list)
    } else {
        let sortedList
        switch (type) {
        case 'duration':
            sortedList = durationSort(list, ascending)
            break
        case 'result':
            sortedList = genericSort(list, type, ascending, initialOrder)
            break
        default:
            sortedList = genericSort(list, type, ascending)
            break
        }
        manager.setRender(sortedList)
    }
}

const doSort = (type, skipDirection) => {
    const newSortType = storageModule.getSort(manager.initialSort) !== type
    const currentAsc = storageModule.getSortDirection()
    let ascending
    if (skipDirection) {
        ascending = currentAsc
    } else {
        ascending = newSortType ? false : !currentAsc
    }
    storageModule.setSort(type)
    storageModule.setSortDirection(ascending)

    const list = manager.testSubset
    const sortedList = type === 'duration' ? durationSort(list, ascending) : genericSort(list, type, ascending)
    manager.setRender(sortedList)
}

module.exports = {
    doInitSort,
    doSort,
}

},{"./datamanager.js":1,"./storage.js":8}],8:[function(require,module,exports){
const possibleFilters = [
    'passed',
    'skipped',
    'failed',
    'error',
    'xfailed',
    'xpassed',
    'rerun',
]

const getVisible = () => {
    const url = new URL(window.location.href)
    const settings = new URLSearchParams(url.search).get('visible')
    const lower = (item) => {
        const lowerItem = item.toLowerCase()
        if (possibleFilters.includes(lowerItem)) {
            return lowerItem
        }
        return null
    }
    return settings === null ?
        possibleFilters :
        [...new Set(settings?.split(',').map(lower).filter((item) => item))]
}

const hideCategory = (categoryToHide) => {
    const url = new URL(window.location.href)
    const visibleParams = new URLSearchParams(url.search).get('visible')
    const currentVisible = visibleParams ? visibleParams.split(',') : [...possibleFilters]
    const settings = [...new Set(currentVisible)].filter((f) => f !== categoryToHide).join(',')

    url.searchParams.set('visible', settings)
    window.history.pushState({}, null, unescape(url.href))
}

const showCategory = (categoryToShow) => {
    if (typeof window === 'undefined') {
        return
    }
    const url = new URL(window.location.href)
    const currentVisible = new URLSearchParams(url.search).get('visible')?.split(',').filter(Boolean) ||
        [...possibleFilters]
    const settings = [...new Set([categoryToShow, ...currentVisible])]
    const noFilter = possibleFilters.length === settings.length || !settings.length

    noFilter ? url.searchParams.delete('visible') : url.searchParams.set('visible', settings.join(','))
    window.history.pushState({}, null, unescape(url.href))
}

const getSort = (initialSort) => {
    const url = new URL(window.location.href)
    let sort = new URLSearchParams(url.search).get('sort')
    if (!sort) {
        sort = initialSort || 'result'
    }
    return sort
}

const setSort = (type) => {
    const url = new URL(window.location.href)
    url.searchParams.set('sort', type)
    window.history.pushState({}, null, unescape(url.href))
}

const getCollapsedCategory = (renderCollapsed) => {
    let categories
    if (typeof window !== 'undefined') {
        const url = new URL(window.location.href)
        const collapsedItems = new URLSearchParams(url.search).get('collapsed')
        switch (true) {
        case !renderCollapsed && collapsedItems === null:
            categories = ['passed']
            break
        case collapsedItems?.length === 0 || /^["']{2}$/.test(collapsedItems):
            categories = []
            break
        case /^all$/.test(collapsedItems) || collapsedItems === null && /^all$/.test(renderCollapsed):
            categories = [...possibleFilters]
            break
        default:
            categories = collapsedItems?.split(',').map((item) => item.toLowerCase()) || renderCollapsed
            break
        }
    } else {
        categories = []
    }
    return categories
}

const getSortDirection = () => JSON.parse(sessionStorage.getItem('sortAsc')) || false
const setSortDirection = (ascending) => sessionStorage.setItem('sortAsc', ascending)

const getCollapsedIds = () => JSON.parse(sessionStorage.getItem('collapsedIds')) || []
const setCollapsedIds = (list) => sessionStorage.setItem('collapsedIds', JSON.stringify(list))

module.exports = {
    getVisible,
    hideCategory,
    showCategory,
    getCollapsedIds,
    setCollapsedIds,
    getSort,
    setSort,
    getSortDirection,
    setSortDirection,
    getCollapsedCategory,
    possibleFilters,
}

},{}]},{},[4]);
    </script>
  </footer>
</html>